<!DOCTYPE html>
<html lang="en"><head>
<script src="quarto_presentation_files/libs/clipboard/clipboard.min.js"></script>
<script src="quarto_presentation_files/libs/quarto-html/tabby.min.js"></script>
<script src="quarto_presentation_files/libs/quarto-html/popper.min.js"></script>
<script src="quarto_presentation_files/libs/quarto-html/tippy.umd.min.js"></script>
<link href="quarto_presentation_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="quarto_presentation_files/libs/quarto-html/light-border.css" rel="stylesheet">
<link href="quarto_presentation_files/libs/quarto-html/quarto-html.min.css" rel="stylesheet" data-mode="light">
<link href="quarto_presentation_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles"><meta charset="utf-8">
  <meta name="generator" content="quarto-1.3.353">

  <meta name="author" content="Weixuan Liu">
  <meta name="author" content="Katerina Kechris">
  <title>SmCCNet 2.0</title>
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, minimal-ui">
  <link rel="stylesheet" href="quarto_presentation_files/libs/revealjs/dist/reset.css">
  <link rel="stylesheet" href="quarto_presentation_files/libs/revealjs/dist/reveal.css">
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    div.columns{display: flex; gap: min(4vw, 1.5em);}
    div.column{flex: auto; overflow-x: auto;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    ul.task-list li input[type="checkbox"] {
      width: 0.8em;
      margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
      vertical-align: middle;
    }
    /* CSS for syntax highlighting */
    pre > code.sourceCode { white-space: pre; position: relative; }
    pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
    pre > code.sourceCode > span:empty { height: 1.2em; }
    .sourceCode { overflow: visible; }
    code.sourceCode > span { color: inherit; text-decoration: inherit; }
    div.sourceCode { margin: 1em 0; }
    pre.sourceCode { margin: 0; }
    @media screen {
    div.sourceCode { overflow: auto; }
    }
    @media print {
    pre > code.sourceCode { white-space: pre-wrap; }
    pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
    }
    pre.numberSource code
      { counter-reset: source-line 0; }
    pre.numberSource code > span
      { position: relative; left: -4em; counter-increment: source-line; }
    pre.numberSource code > span > a:first-child::before
      { content: counter(source-line);
        position: relative; left: -1em; text-align: right; vertical-align: baseline;
        border: none; display: inline-block;
        -webkit-touch-callout: none; -webkit-user-select: none;
        -khtml-user-select: none; -moz-user-select: none;
        -ms-user-select: none; user-select: none;
        padding: 0 4px; width: 4em;
        color: #aaaaaa;
      }
    pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
    div.sourceCode
      { color: #003b4f; background-color: #f1f3f5; }
    @media screen {
    pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
    }
    code span { color: #003b4f; } /* Normal */
    code span.al { color: #ad0000; } /* Alert */
    code span.an { color: #5e5e5e; } /* Annotation */
    code span.at { color: #657422; } /* Attribute */
    code span.bn { color: #ad0000; } /* BaseN */
    code span.bu { } /* BuiltIn */
    code span.cf { color: #003b4f; } /* ControlFlow */
    code span.ch { color: #20794d; } /* Char */
    code span.cn { color: #8f5902; } /* Constant */
    code span.co { color: #5e5e5e; } /* Comment */
    code span.cv { color: #5e5e5e; font-style: italic; } /* CommentVar */
    code span.do { color: #5e5e5e; font-style: italic; } /* Documentation */
    code span.dt { color: #ad0000; } /* DataType */
    code span.dv { color: #ad0000; } /* DecVal */
    code span.er { color: #ad0000; } /* Error */
    code span.ex { } /* Extension */
    code span.fl { color: #ad0000; } /* Float */
    code span.fu { color: #4758ab; } /* Function */
    code span.im { color: #00769e; } /* Import */
    code span.in { color: #5e5e5e; } /* Information */
    code span.kw { color: #003b4f; } /* Keyword */
    code span.op { color: #5e5e5e; } /* Operator */
    code span.ot { color: #003b4f; } /* Other */
    code span.pp { color: #ad0000; } /* Preprocessor */
    code span.sc { color: #5e5e5e; } /* SpecialChar */
    code span.ss { color: #20794d; } /* SpecialString */
    code span.st { color: #20794d; } /* String */
    code span.va { color: #111111; } /* Variable */
    code span.vs { color: #20794d; } /* VerbatimString */
    code span.wa { color: #5e5e5e; font-style: italic; } /* Warning */
  </style>
  <link rel="stylesheet" href="quarto_presentation_files/libs/revealjs/dist/theme/quarto.css">
  <link rel="stylesheet" href="styles.css">
  <link href="quarto_presentation_files/libs/revealjs/plugin/quarto-line-highlight/line-highlight.css" rel="stylesheet">
  <link href="quarto_presentation_files/libs/revealjs/plugin/reveal-menu/menu.css" rel="stylesheet">
  <link href="quarto_presentation_files/libs/revealjs/plugin/reveal-menu/quarto-menu.css" rel="stylesheet">
  <link href="quarto_presentation_files/libs/revealjs/plugin/reveal-chalkboard/font-awesome/css/all.css" rel="stylesheet">
  <link href="quarto_presentation_files/libs/revealjs/plugin/reveal-chalkboard/style.css" rel="stylesheet">
  <link href="quarto_presentation_files/libs/revealjs/plugin/quarto-support/footer.css" rel="stylesheet">
  <style type="text/css">

  .callout {
    margin-top: 1em;
    margin-bottom: 1em;  
    border-radius: .25rem;
  }

  .callout.callout-style-simple { 
    padding: 0em 0.5em;
    border-left: solid #acacac .3rem;
    border-right: solid 1px silver;
    border-top: solid 1px silver;
    border-bottom: solid 1px silver;
    display: flex;
  }

  .callout.callout-style-default {
    border-left: solid #acacac .3rem;
    border-right: solid 1px silver;
    border-top: solid 1px silver;
    border-bottom: solid 1px silver;
  }

  .callout .callout-body-container {
    flex-grow: 1;
  }

  .callout.callout-style-simple .callout-body {
    font-size: 1rem;
    font-weight: 400;
  }

  .callout.callout-style-default .callout-body {
    font-size: 0.9rem;
    font-weight: 400;
  }

  .callout.callout-titled.callout-style-simple .callout-body {
    margin-top: 0.2em;
  }

  .callout:not(.callout-titled) .callout-body {
      display: flex;
  }

  .callout:not(.no-icon).callout-titled.callout-style-simple .callout-content {
    padding-left: 1.6em;
  }

  .callout.callout-titled .callout-header {
    padding-top: 0.2em;
    margin-bottom: -0.2em;
  }

  .callout.callout-titled .callout-title  p {
    margin-top: 0.5em;
    margin-bottom: 0.5em;
  }
    
  .callout.callout-titled.callout-style-simple .callout-content  p {
    margin-top: 0;
  }

  .callout.callout-titled.callout-style-default .callout-content  p {
    margin-top: 0.7em;
  }

  .callout.callout-style-simple div.callout-title {
    border-bottom: none;
    font-size: .9rem;
    font-weight: 600;
    opacity: 75%;
  }

  .callout.callout-style-default  div.callout-title {
    border-bottom: none;
    font-weight: 600;
    opacity: 85%;
    font-size: 0.9rem;
    padding-left: 0.5em;
    padding-right: 0.5em;
  }

  .callout.callout-style-default div.callout-content {
    padding-left: 0.5em;
    padding-right: 0.5em;
  }

  .callout.callout-style-simple .callout-icon::before {
    height: 1rem;
    width: 1rem;
    display: inline-block;
    content: "";
    background-repeat: no-repeat;
    background-size: 1rem 1rem;
  }

  .callout.callout-style-default .callout-icon::before {
    height: 0.9rem;
    width: 0.9rem;
    display: inline-block;
    content: "";
    background-repeat: no-repeat;
    background-size: 0.9rem 0.9rem;
  }

  .callout-title {
    display: flex
  }
    
  .callout-icon::before {
    margin-top: 1rem;
    padding-right: .5rem;
  }

  .callout.no-icon::before {
    display: none !important;
  }

  .callout.callout-titled .callout-body > .callout-content > :last-child {
    margin-bottom: 0.5rem;
  }

  .callout.callout-titled .callout-icon::before {
    margin-top: .5rem;
    padding-right: .5rem;
  }

  .callout:not(.callout-titled) .callout-icon::before {
    margin-top: 1rem;
    padding-right: .5rem;
  }

  /* Callout Types */

  div.callout-note {
    border-left-color: #4582ec !important;
  }

  div.callout-note .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAEU0lEQVRYCcVXTWhcVRQ+586kSUMMxkyaElstCto2SIhitS5Ek8xUKV2poatCcVHtUlFQk8mbaaziwpWgglJwVaquitBOfhQXFlqlzSJpFSpIYyXNjBNiTCck7x2/8/LeNDOZxDuEkgOXe++553zfefee+/OYLOXFk3+1LLrRdiO81yNqZ6K9cG0P3MeFaMIQjXssE8Z1JzLO9ls20MBZX7oG8w9GxB0goaPrW5aNMp1yOZIa7Wv6o2ykpLtmAPs/vrG14Z+6d4jpbSKuhdcSyq9wGMPXjonwmESXrriLzFGOdDBLB8Y6MNYBu0dRokSygMA/mrun8MGFN3behm6VVAwg4WR3i6FvYK1T7MHo9BK7ydH+1uurECoouk5MPRyVSBrBHMYwVobG2aOXM07sWrn5qgB60rc6mcwIDJtQrnrEr44kmy+UO9r0u9O5/YbkS9juQckLed3DyW2XV/qWBBB3ptvI8EUY3I9p/67OW+g967TNr3Sotn3IuVlfMLVnsBwH4fsnebJvyGm5GeIUA3jljERmrv49SizPYuq+z7c2H/jlGC+Ghhupn/hcapqmcudB9jwJ/3jvnvu6vu5lVzF1fXyZuZZ7U8nRmVzytvT+H3kilYvH09mLWrQdwFSsFEsxFVs5fK7A0g8gMZjbif4ACpKbjv7gNGaD8bUrlk8x+KRflttr22JEMRUbTUwwDQScyzPgedQHZT0xnx7ujw2jfVfExwYHwOsDTjLdJ2ebmeQIlJ7neo41s/DrsL3kl+W2lWvAga0tR3zueGr6GL78M3ifH0rGXrBC2aAR8uYcIA5gwV8zIE8onoh8u0Fca/ciF7j1uOzEnqcIm59sEXoGc0+z6+H45V1CvAvHcD7THztu669cnp+L0okAeIc6zjbM/24LgGM1gZk7jnRu1aQWoU9sfUOuhrmtaPIO3YY1KLLWZaEO5TKUbMY5zx8W9UJ6elpLwKXbsaZ4EFl7B4bMtDv0iRipKoDQT2sNQI9b1utXFdYisi+wzZ/ri/1m7QfDgEuvgUUEIJPq3DhX/5DWNqIXDOweC2wvIR90Oq3lDpdMIgD2r0dXvGdsEW5H6x6HLRJYU7C69VefO1x8Gde1ZFSJLfWS1jbCnhtOPxmpfv2LXOA2Xk2tvnwKKPFuZ/oRmwBwqRQDcKNeVQkYcOjtWVBuM/JuYw5b6isojIkYxyYAFn5K7ZBF10fea52y8QltAg6jnMqNHFBmGkQ1j+U43HMi2xMar1Nv0zGsf1s8nUsmUtPOOrbFIR8bHFDMB5zL13Gmr/kGlCkUzedTzzmzsaJXhYawnA3UmARpiYj5ooJZiUoxFRtK3X6pgNPv+IZVPcnwbOl6f+aBaO1CNvPW9n9LmCp01nuSaTRF2YxHqZ8DYQT6WsXT+RD6eUztwYLZ8rM+rcPxamv1VQzFUkzFXvkiVrySGQgJNvXHJAxiU3/NwiC03rSf05VBaPtu/Z7/B8Yn/w7eguloAAAAAElFTkSuQmCC');
  }

  div.callout-note.callout-style-default .callout-title {
    background-color: #dae6fb
  }

  div.callout-important {
    border-left-color: #d9534f !important;
  }

  div.callout-important .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAEKklEQVRYCcVXTWhcVRS+575MJym48A+hSRFr00ySRQhURRfd2HYjk2SSTokuBCkU2o0LoSKKraKIBTcuFCoidGFD08nkBzdREbpQ1EDNIv8qSGMFUboImMSZd4/f9zJv8ibJMC8xJQfO3HPPPef7zrvvvnvviIkpC9nsw0UttFunbUhpFzFtarSd6WJkStVMw5xyVqYTvkwfzuf/5FgtkVoB0729j1rjXwThS7Vio+Mo6DNnvLfahoZ+i/o32lULuJ3NNiz7q6+pyAUkJaFF6JwaM2lUJlV0MlnQn5aTRbEu0SEqHUa0A4AdiGuB1kFXRfVyg5d87+Dg4DL6m2TLAub60ilj7A1Ec4odSAc8X95sHh7+ZRPCFo6Fnp7HfU/fBng/hi10CjCnWnJjsxvDNxWw0NfV6Rv5GgP3I3jGWXumdTD/3cbEOP2ZbOZp69yniG3FQ9z1jD7bnBu9Fc2tKGC2q+uAJOQHBDRiZX1x36o7fWBs7J9ownbtO+n0/qWkvW7UPIfc37WgT6ZGR++EOJyeQDSb9UB+DZ1G6DdLDzyS+b/kBCYGsYgJbSQHuThGKRcw5xdeQf8YdNHsc6ePXrlSYMBuSIAFTGAtQo+VuALo4BX83N190NWZWbynBjhOHsmNfFWLeL6v+ynsA58zDvvAC8j5PkbOcXCMg2PZFk3q8MjI7WAG/Dp9AwP7jdGBOOQkAvlFUB+irtm16I1Zw9YBcpGTGXYmk3kQIC/Cds55l+iMI3jqhjAuaoe+am2Jw5GT3Nbz3CkE12NavmzN5+erJW7046n/CH1RO/RVa8lBLozXk9uqykkGAyRXLWlLv5jyp4RFsG5vGVzpDLnIjTWgnRy2Rr+tDKvRc7Y8AyZq10jj8DqXdnIRNtFZb+t/ZRtXcDiVnzpqx8mPcDWxgARUqx0W1QB9MeUZiNrV4qP+Ehc+BpNgATsTX8ozYKL2NtFYAHc84fG7ndxUPr+AR/iQSns7uSUufAymwDOb2+NjK27lEFocm/EE2WpyIy/Hi66MWuMKJn8RvxIcj87IM5Vh9663ziW36kR0HNenXuxmfaD8JC7tfKbrhFr7LiZCrMjrzTeGx+PmkosrkNzW94ObzwocJ7A1HokLolY+AvkTiD/q1H0cN48c5EL8Crkttsa/AXQVDmutfyku0E7jShx49XqV3MFK8IryDhYVbj7Sj2P2eBxwcXoe8T8idsKKPRcnZw1b+slFTubwUwhktrfnAt7J++jwQtLZcm3sr9LQrjRzz6cfMv9aLvgmnAGvpoaGLxM4mAEaLV7iAzQ3oU0IvD5x9ix3yF2RAAuYAOO2f7PEFWCXZ4C9Pb2UsgDeVnFSpbFK7/IWu7TPTvBqzbGdCHOJQSxiEjt6IyZmxQyEJHv6xyQsYk//moVFsN2zP6fRImjfq7/n/wFDguUQFNEwugAAAABJRU5ErkJggg==');
  }

  div.callout-important.callout-style-default .callout-title {
    background-color: #f7dddc
  }

  div.callout-warning {
    border-left-color: #f0ad4e !important;
  }

  div.callout-warning .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAETklEQVRYCeVWW2gcVRg+58yaTUnizqbipZeX4uWhBEniBaoUX1Ioze52t7sRq6APio9V9MEaoWlVsFasRq0gltaAPuxms8lu0gcviE/FFOstVbSIxgcv6SU7EZqmdc7v9+9mJtNks51NTUH84ed889/PP+cmxP+d5FIbMJmNbpREu4WUkiTtCicKny0l1pIKmBzovF2S+hIJHX8iEu3hZJ5lNZGqyRrGSIQpq15AzF28jgpeY6yk6GVdrfFqdrD6Iw+QlB8g0YS2g7dyQmXM/IDhBhT0UCiRf59lfqmmDvzRt6kByV/m4JjtzuaujMUM2c5Z2d6JdKrRb3K2q6mA+oYVz8JnDdKPmmNthzkAk/lN63sYPgevrguc72aZX/L9C6x09GYyxBgCX4NlvyGUHOKELlm5rXeR1kchuChJt4SSwyddZRXgvwMGvYo4QSlk3/zkHD8UHxwVJA6zjZZqP8v8kK8OWLnIZtLyCAJagYC4rTGW/9Pqj92N/c+LUaAj27movwbi19tk/whRCIE7Q9vyI6yvRpftAKVTdUjOW40X3h5OXsKCdmFcx0xlLJoSuQngnrJe7Kcjm4OMq9FlC7CMmScQANuNvjfP3PjGXDBaUQmbp296S5L4DrpbrHN1T87ZVEZVCzg1FF0Ft+dKrlLukI+/c9ENo+TvlTDbYFvuKPtQ9+l052rXrgKoWkDAFnvh0wTOmYn8R5f4k/jN/fZiCM1tQx9jQQ4ANhqG4hiL0qIFTGViG9DKB7GYzgubnpofgYRwO+DFjh0Zin2m4b/97EDkXkc+f6xYAPX0KK2I/7fUQuwzuwo/L3AkcjugPNixC8cHf0FyPjWlItmLxWw4Ou9YsQCr5fijMGoD/zpdRy95HRysyXA74MWOnscpO4j2y3HAVisw85hX5+AFBRSHt4ShfLFkIMXTqyKFc46xdzQM6XbAi702a7sy04J0+feReMFKp5q9esYLCqAZYw/k14E/xcLLsFElaornTuJB0svMuJINy8xkIYuL+xPAlWRceH6+HX7THJ0djLUom46zREu7tTkxwmf/FdOZ/sh6Q8qvEAiHpm4PJ4a/doJe0gH1t+aHRgCzOvBvJedEK5OFE5jpm4AGP2a8Dxe3gGJ/pAutug9Gp6he92CsSsWBaEcxGx0FHytmIpuqGkOpldqNYQK8cSoXvd+xLxXADw0kf6UkJNFtdo5MOgaLjiQOQHcn+A6h5NuL2s0qsC2LOM75PcF3yr5STuBSAcGG+meA14K/CI21HcS4LBT6tv0QAh8Dr5l93AhZzG5ZJ4VxAqdZUEl9z7WJ4aN+svMvwHHL21UKTd1mqvChH7/Za5xzXBBKrUcB0TQ+Ulgkfbi/H/YT5EptrGzsEK7tR1B7ln9BBwckYfMiuSqklSznIuoIIOM42MQO+QnduCoFCI0bpkzjCjddHPN/F+2Yu+sd9bKNpVwHhbS3LluK/0zgfwD0xYI5dXuzlQAAAABJRU5ErkJggg==');
  }

  div.callout-warning.callout-style-default .callout-title {
    background-color: #fcefdc
  }

  div.callout-tip {
    border-left-color: #02b875 !important;
  }

  div.callout-tip .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAADr0lEQVRYCe1XTWgTQRj9ZjZV8a9SPIkKgj8I1bMHsUWrqYLVg4Ue6v9BwZOxSYsIerFao7UiUryIqJcqgtpimhbBXoSCVxUFe9CTiogUrUp2Pt+3aUI2u5vdNh4dmMzOzHvvezuz8xNFM0mjnbXaNu1MvFWRXkXEyE6aYOYJpdW4IXuA4r0fo8qqSMDBU0v1HJUgVieAXxzCsdE/YJTdFcVIZQNMyhruOMJKXYFoLfIfIvVIMWdsrd+Rpd86ZmyzzjJmLStqRn0v8lzkb4rVIXvnpScOJuAn2ACC65FkPzEdEy4TPWRLJ2h7z4cArXzzaOdKlbOvKKX25Wl00jSnrwVxAg3o4dRxhO13RBSdNvH0xSARv3adTXbBdTf64IWO2vH0LT+cv4GR1DJt+DUItaQogeBX/chhbTBxEiZ6gftlDNXTrvT7co4ub5A6gp9HIcHvzTa46OS5fBeP87Qm0fQkr4FsYgVQ7Qg+ZayaDg9jhg1GkWj8RG6lkeSacrrHgDaxdoBiZPg+NXV/KifMuB6//JmYH4CntVEHy/keA6x4h4CU5oFy8GzrBS18cLJMXcljAKB6INjWsRcuZBWVaS3GDrqB7rdapVIeA+isQ57Eev9eCqzqOa81CY05VLd6SamW2wA2H3SiTbnbSxmzfp7WtKZkqy4mdyAlGx7ennghYf8voqp9cLSgKdqNfa6RdRsAAkPwRuJZNbpByn+RrJi1RXTwdi8RQF6ymDwGMAtZ6TVE+4uoKh+MYkcLsT0Hk8eAienbiGdjJHZTpmNjlbFJNKDVAp2fJlYju6IreQxQ08UJDNYdoLSl6AadO+fFuCQqVMB1NJwPm69T04Wv5WhfcWyfXQB+wXRs1pt+nCknRa0LVzSA/2B+a9+zQJadb7IyyV24YAxKp2Jqs3emZTuNnKxsah+uabKbMk7CbTgJx/zIgQYErIeTKRQ9yD9wxVof5YolPHqaWo7TD6tJlh7jQnK5z2n3+fGdggIOx2kaa2YI9QWarc5Ce1ipNWMKeSG4DysFF52KBmTNMmn5HqCFkwy34rDg05gDwgH3bBi+sgFhN/e8QvRn8kbamCOhgrZ9GJhFDgfcMHzFb6BAtjKpFhzTjwv1KCVuxHvCbsSiEz4CANnj84cwHdFXAbAOJ4LTSAawGWFn5tDhLMYz6nWeU2wJfIhmIJBefcd/A5FWQWGgrWzyORZ3Q6HuV+Jf0Bj+BTX69fm1zWgK7By1YTXchFDORywnfQ7GpzOo6S+qECrsx2ifVQAAAABJRU5ErkJggg==');
  }

  div.callout-tip.callout-style-default .callout-title {
    background-color: #ccf1e3
  }

  div.callout-caution {
    border-left-color: #fd7e14 !important;
  }

  div.callout-caution .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAACV0lEQVRYCdVWzWoUQRCuqp2ICBLJXgITZL1EfQDBW/bkzUMUD7klD+ATSHBEfAIfQO+iXsWDxJsHL96EHAwhgzlkg8nBg25XWb0zIb0zs9muYYWkoKeru+vn664fBqElyZNuyh167NXJ8Ut8McjbmEraKHkd7uAnAFku+VWdb3reSmRV8PKSLfZ0Gjn3a6Xlcq9YGb6tADjn+lUfTXtVmaZ1KwBIvFI11rRXlWlatwIAAv2asaa9mlB9wwygiDX26qaw1yYPzFXg2N1GgG0FMF8Oj+VIx7E/03lHx8UhvYyNZLN7BwSPgekXXLribw7w5/c8EF+DBK5idvDVYtEEwMeYefjjLAdEyQ3M9nfOkgnPTEkYU+sxMq0BxNR6jExrAI31H1rzvLEfRIdgcv1XEdj6QTQAS2wtstEALLG1yEZ3QhH6oDX7ExBSFEkFINXH98NTrme5IOaaA7kIfiu2L8A3qhH9zRbukdCqdsA98TdElyeMe5BI8Rs2xHRIsoTSSVFfCFCWGPn9XHb4cdobRIWABNf0add9jakDjQJpJ1bTXOJXnnRXHRf+dNL1ZV1MBRCXhMbaHqGI1JkKIL7+i8uffuP6wVQAzO7+qVEbF6NbS0LJureYcWXUUhH66nLR5rYmva+2tjRFtojkM2aD76HEGAD3tPtKM309FJg5j/K682ywcWJ3PASCcycH/22u+Bh7Aa0ehM2Fu4z0SAE81HF9RkB21c5bEn4Dzw+/qNOyXr3DCTQDMBOdhi4nAgiFDGCinIa2owCEChUwD8qzd03PG+qdW/4fDzjUMcE1ZpIAAAAASUVORK5CYII=');
  }

  div.callout-caution.callout-style-default .callout-title {
    background-color: #ffe5d0
  }

  </style>
  <style type="text/css">
    .reveal div.sourceCode {
      margin: 0;
      overflow: auto;
    }
    .reveal div.hanging-indent {
      margin-left: 1em;
      text-indent: -1em;
    }
    .reveal .slide:not(.center) {
      height: 100%;
    }
    .reveal .slide.scrollable {
      overflow-y: auto;
    }
    .reveal .footnotes {
      height: 100%;
      overflow-y: auto;
    }
    .reveal .slide .absolute {
      position: absolute;
      display: block;
    }
    .reveal .footnotes ol {
      counter-reset: ol;
      list-style-type: none; 
      margin-left: 0;
    }
    .reveal .footnotes ol li:before {
      counter-increment: ol;
      content: counter(ol) ". "; 
    }
    .reveal .footnotes ol li > p:first-child {
      display: inline-block;
    }
    .reveal .slide ul,
    .reveal .slide ol {
      margin-bottom: 0.5em;
    }
    .reveal .slide ul li,
    .reveal .slide ol li {
      margin-top: 0.4em;
      margin-bottom: 0.2em;
    }
    .reveal .slide ul[role="tablist"] li {
      margin-bottom: 0;
    }
    .reveal .slide ul li > *:first-child,
    .reveal .slide ol li > *:first-child {
      margin-block-start: 0;
    }
    .reveal .slide ul li > *:last-child,
    .reveal .slide ol li > *:last-child {
      margin-block-end: 0;
    }
    .reveal .slide .columns:nth-child(3) {
      margin-block-start: 0.8em;
    }
    .reveal blockquote {
      box-shadow: none;
    }
    .reveal .tippy-content>* {
      margin-top: 0.2em;
      margin-bottom: 0.7em;
    }
    .reveal .tippy-content>*:last-child {
      margin-bottom: 0.2em;
    }
    .reveal .slide > img.stretch.quarto-figure-center,
    .reveal .slide > img.r-stretch.quarto-figure-center {
      display: block;
      margin-left: auto;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-left,
    .reveal .slide > img.r-stretch.quarto-figure-left  {
      display: block;
      margin-left: 0;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-right,
    .reveal .slide > img.r-stretch.quarto-figure-right  {
      display: block;
      margin-left: auto;
      margin-right: 0; 
    }
  </style>
</head>
<body class="quarto-light">
  <div class="reveal">
    <div class="slides">

<section id="title-slide" class="quarto-title-block center">
  <h1 class="title">SmCCNet 2.0</h1>
  <p class="subtitle">A New Generation of SmCCNet Package for Multi-omics Network Inference</p>

<div class="quarto-title-authors">
<div class="quarto-title-author">
<div class="quarto-title-author-name">
Weixuan Liu 
</div>
<div class="quarto-title-author-email">
<a href="mailto:weixuan.liu@cuanschutz.edu">weixuan.liu@cuanschutz.edu</a>
</div>
</div>
<div class="quarto-title-author">
<div class="quarto-title-author-name">
Katerina Kechris 
</div>
</div>
</div>

</section>
<section id="smccnet-description-file" class="slide level2 scrollable" data-transition="fade">
<h2>SmCCNet Description File</h2>

<img data-src="images/packageDescription.png" class="r-stretch"></section>
<section id="smccnet-1.0-workflow" class="slide level2 scrollable" data-transition="fade">
<h2>SmCCNet 1.0 Workflow</h2>

<img data-src="images/SmCCNetWorkflow.png" class="r-stretch"></section>
<section id="drawbacks-of-smccnet-1.0" class="slide level2 smaller scrollable" data-transition="fade" data-auto-animate="true">
<h2 data-id="quarto-animate-title">Drawbacks of SmCCNet 1.0</h2>
<p>Multiple Drawbacks are found in SmCCNet 1.0.0</p>
<div>
<ul>
<li class="fragment">Can’t handle single-omics data.</li>
<li class="fragment">Can’t handle more than two omics data.</li>
<li class="fragment">Computationally intensive.</li>
<li class="fragment">Long code for pipeline setup.</li>
<li class="fragment">No network pruning and NetSHy summarization score.</li>
</ul>
</div>
</section>
<section id="updates-of-smccnet-2.0" class="slide level2 smaller scrollable" data-transition="fade" data-auto-animate="true">
<h2 data-id="quarto-animate-title">Updates of SmCCNet 2.0</h2>
<p>There are multiple updates in SmCCNet 2.0.0:</p>
<div>
<ul>
<li class="fragment">Single/Multi-omics SmCCNet with quantitative phenotype (generalizable).</li>
<li class="fragment">Single/Multi-omics SmCCNet with binary phenotype.</li>
<li class="fragment">Network pruning algorithm.</li>
<li class="fragment">Fast automated SmCCNet.</li>
</ul>
</div>
<div class="fragment">
<p><br> …and more</p>
</div>
</section>
<section id="smccnet-2.0-workflow" class="slide level2 scrollable" data-transition="fade">
<h2>SmCCNet 2.0 Workflow</h2>

<img data-src="images/2ndSmCCNet.drawio.png" class="r-stretch"></section>
<section id="functions-in-smccnet-1.0.0" class="slide level2 smaller scrollable" data-transition="fade">
<h2>Functions in SmCCNet 1.0.0</h2>
<p>Everything except for <code>getAbar()</code> are retired.</p>
<table>
<colgroup>
<col style="width: 29%">
<col style="width: 70%">
</colgroup>
<thead>
<tr class="header">
<th>Function</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>getRobustPseudoWeights()</code></td>
<td>Compute aggregated (SmCCA) canonical weights.</td>
</tr>
<tr class="even">
<td><code>getAbar()</code></td>
<td>Calculate similarity matrix based on canonical weights.</td>
</tr>
<tr class="odd">
<td><code>getMultiOmicsModules()</code></td>
<td>Perform hierarchical tree cutting on the similarity matrix and extract clades with multi-omics features.</td>
</tr>
<tr class="even">
<td><code>plotMultiOmicsNetwork()</code></td>
<td>Plot (pruned or full) multi-omics subnetworks.</td>
</tr>
</tbody>
</table>
</section>
<section id="new-functions-in-smccnet-2.0.0" class="slide level2 smaller scrollable" data-transition="fade">
<h2>New Functions in SmCCNet 2.0.0</h2>
<table>
<colgroup>
<col style="width: 30%">
<col style="width: 69%">
</colgroup>
<thead>
<tr class="header">
<th>Function</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>aggregateCVSingle()</code></td>
<td>Save cross-validation result and recommend penalty term selection.</td>
</tr>
<tr class="even">
<td><code>classifierEval()</code></td>
<td>Evaluate binary classifier’s performance using various metrics.</td>
</tr>
<tr class="odd">
<td><code>dataPreprocess()</code></td>
<td>Preprocess data for SmCCNet (centering, scaling, filtering, regressing out covariates).</td>
</tr>
<tr class="even">
<td><code>fastAutoSmCCNet()</code></td>
<td>Automated SmCCNet: identifies project type, preprocesses data, and runs the SmCCNet pipeline.</td>
</tr>
<tr class="odd">
<td><code>getCanWeightsMulti()</code></td>
<td>Run SmCCA and return canonical weight.</td>
</tr>
<tr class="even">
<td><code>getCanCorMulti()</code></td>
<td>Retrieve canonical correlation value for SmCCA.</td>
</tr>
<tr class="odd">
<td><code>getRobustWeightsMulti()</code></td>
<td>Calculate canonical weights for SmCCA.</td>
</tr>
<tr class="even">
<td><code>getRobustWeightsMultiBinary()</code></td>
<td>SmCCNet for multi-omics data with binary phenotype using SmCCA and SPLSDA.</td>
</tr>
<tr class="odd">
<td><code>getRobustWeightsSingle()</code></td>
<td>Compute SmCCA canonical weights for single omics with quantitative phenotype.</td>
</tr>
<tr class="even">
<td><code>getRobustWeightsSingleBinary()</code></td>
<td>Compute SmCCA canonical weights for single omics with binary phenotype.</td>
</tr>
<tr class="odd">
<td><code>getOmicsModules()</code></td>
<td>Extract clades with omics features from similarity matrix.</td>
</tr>
<tr class="even">
<td><code>networkPruning()</code></td>
<td>Extract summarization scores for network module and rank omics features.</td>
</tr>
<tr class="odd">
<td><code>scalingFactorInput()</code></td>
<td>Input annotation of omics data and provide scaling factor for SmCCNet.</td>
</tr>
</tbody>
</table>
</section>
<section id="example-multi-omics-with-quantitative-phenotype" class="slide level2 scrollable" data-transition="fade">
<h2>Example: Multi-omics with Quantitative Phenotype</h2>
<div>
<ul>
<li class="fragment">Step 1: Preprocessing</li>
<li class="fragment">Step 2: Determine optimal sparsity parameters through cross-validation</li>
<li class="fragment">Step 3: Run SmCCA with pre-selected penalty terms</li>
<li class="fragment">Step 4: Obtain multi-omics modules through network clustering</li>
<li class="fragment">Step 5: Network pruning and summarization</li>
<li class="fragment">Step 6: Network visualization</li>
</ul>
</div>
</section>
<section id="step-1-preprocessing-new-feature" class="slide level2 smaller scrollable" data-transition="fade" data-auto-animate="true">
<h2 data-id="quarto-animate-title">Step 1: Preprocessing (new feature)</h2>
<div>
<ul>
<li class="fragment">SmCCNet 1.0.0: Preprocess data manually.</li>
<li class="fragment">SmCCNet 2.0.0: Preprocess data with function.</li>
</ul>
</div>
<div class="fragment">
<p><br></p>
<table>
<colgroup>
<col style="width: 29%">
<col style="width: 70%">
</colgroup>
<thead>
<tr class="header">
<th>Methods</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>CoV Filtering</code></td>
<td>Filter features using coefficients of variation (CoV).</td>
</tr>
<tr class="even">
<td><code>Standardization</code></td>
<td>Center and/or scale data.</td>
</tr>
<tr class="odd">
<td><code>Adjust for Covariates</code></td>
<td>Regress out specified covariates and provide residuals.</td>
</tr>
</tbody>
</table>
</div>
<div class="fragment">
<div class="sourceCode" id="cb1" data-code-line-numbers="1-2|3-5|6|7|8|9-11"><pre class="sourceCode numberSource r number-lines code-with-copy" data-id="quarto-animate-code"><code class="sourceCode r hljs"><span id="cb1-1" class="hljs-ln-code"><a href="#cb1-1"></a><span class="co"># create data list</span></span>
<span id="cb1-2" class="hljs-ln-code"><a href="#cb1-2"></a>X <span class="ot">&lt;-</span> <span class="fu">list</span>(X1, X2)</span>
<span id="cb1-3" class="hljs-ln-code"><a href="#cb1-3"></a><span class="co"># preprocess data</span></span>
<span id="cb1-4" class="hljs-ln-code"><a href="#cb1-4"></a>processed_data <span class="ot">&lt;-</span> <span class="fu">lapply</span>(X, <span class="cf">function</span>(Xmatrix){</span>
<span id="cb1-5" class="hljs-ln-code"><a href="#cb1-5"></a>  <span class="fu">as.matrix</span>(<span class="fu">dataPreprocess</span>(<span class="at">X =</span> <span class="fu">as.data.frame</span>(Xmatrix), </span>
<span id="cb1-6" class="hljs-ln-code"><a href="#cb1-6"></a>                           <span class="at">covariates =</span> <span class="cn">NULL</span>, </span>
<span id="cb1-7" class="hljs-ln-code"><a href="#cb1-7"></a>                           <span class="at">is_cv =</span> <span class="cn">TRUE</span>,</span>
<span id="cb1-8" class="hljs-ln-code"><a href="#cb1-8"></a>                           <span class="at">cv_quantile =</span> <span class="fl">0.2</span>,</span>
<span id="cb1-9" class="hljs-ln-code"><a href="#cb1-9"></a>                           <span class="at">center =</span> <span class="cn">TRUE</span>, </span>
<span id="cb1-10" class="hljs-ln-code"><a href="#cb1-10"></a>                           <span class="at">scale =</span> <span class="cn">TRUE</span>))</span>
<span id="cb1-11" class="hljs-ln-code"><a href="#cb1-11"></a>                 })</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="step-2-determine-optimal-sparsity-parameters-through-cross-validation-upgraded" class="slide level2 smaller scrollable" data-transition="fade" data-auto-animate="true">
<h2 data-id="quarto-animate-title">Step 2: Determine optimal sparsity parameters through cross-validation (upgraded)</h2>
<div>
<ul>
<li class="fragment">SmCCNet 1.0.0: For two-omics only.</li>
<li class="fragment">SmCCNet 2.0.0: Generalizable to more than two omics data.</li>
</ul>
</div>
<div class="fragment">
<p><br></p>
<table>
<colgroup>
<col style="width: 27%">
<col style="width: 72%">
</colgroup>
<thead>
<tr class="header">
<th>Arguments</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>K</code></td>
<td>Number of folds in CV. Suitable for 5-fold CV. For training sets with &lt;30 samples or unscalable sets, choose a smaller <span class="math inline">\(K\)</span>.</td>
</tr>
<tr class="even">
<td><code>CCcoef</code></td>
<td>Coefficients <span class="math inline">\((a_{i,j}, b_{i})\)</span> for weighted SmCCA. Can be supplied manually or iteratively.</td>
</tr>
<tr class="odd">
<td><code>PenExpand</code></td>
<td>Penalty option matrix for <span class="math inline">\(X_1, X_2\)</span>. Each row is a penalty option pair <span class="math inline">\(l = (l_1, l_2)\)</span> with <span class="math inline">\(0&lt;l_1, l_2&lt;1\)</span>.</td>
</tr>
<tr class="even">
<td><code>X</code></td>
<td>List of omics data with consistent set and order of subjects.</td>
</tr>
<tr class="odd">
<td><code>num_omics</code></td>
<td>Total number of omics data in <span class="math inline">\(X\)</span>.</td>
</tr>
<tr class="even">
<td><code>tuneLength</code></td>
<td>Total number of candidate penalty terms for each penalty parameter.</td>
</tr>
<tr class="odd">
<td><code>minTune</code></td>
<td>Minimum possible value for penalty parameters.</td>
</tr>
<tr class="even">
<td><code>maxTune</code></td>
<td>Maximum possible tuning value for penalty parameters.</td>
</tr>
</tbody>
</table>
</div>
</section>
<section id="parameter-setup-example" class="slide level2 smaller scrollable" data-transition="fade" data-auto-animate="true">
<h2 data-id="quarto-animate-title">Parameter setup example</h2>
<div class="sourceCode" id="cb2"><pre class="sourceCode numberSource r number-lines code-with-copy" data-id="quarto-animate-code"><code class="sourceCode r hljs"><span id="cb2-1" class="hljs-ln-code"><a href="#cb2-1"></a><span class="fu">library</span>(SmCCNet)</span>
<span id="cb2-2" class="hljs-ln-code"><a href="#cb2-2"></a><span class="co"># import data</span></span>
<span id="cb2-3" class="hljs-ln-code"><a href="#cb2-3"></a><span class="fu">data</span>(ExampleData)</span>
<span id="cb2-4" class="hljs-ln-code"><a href="#cb2-4"></a>p1 <span class="ot">&lt;-</span> <span class="fu">ncol</span>(X1)</span>
<span id="cb2-5" class="hljs-ln-code"><a href="#cb2-5"></a>p2 <span class="ot">&lt;-</span> <span class="fu">ncol</span>(X2)</span>
<span id="cb2-6" class="hljs-ln-code"><a href="#cb2-6"></a>n <span class="ot">&lt;-</span> <span class="fu">nrow</span>(X1)</span>
<span id="cb2-7" class="hljs-ln-code"><a href="#cb2-7"></a>AbarLabel <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">colnames</span>(<span class="fu">cbind</span>(X1, X2)))</span>
<span id="cb2-8" class="hljs-ln-code"><a href="#cb2-8"></a><span class="co"># number of folds in K-fold CV.</span></span>
<span id="cb2-9" class="hljs-ln-code"><a href="#cb2-9"></a>K <span class="ot">&lt;-</span> <span class="dv">3</span> </span>
<span id="cb2-10" class="hljs-ln-code"><a href="#cb2-10"></a>N <span class="ot">&lt;-</span> <span class="fu">nrow</span>(X1)</span>
<span id="cb2-11" class="hljs-ln-code"><a href="#cb2-11"></a><span class="co"># create a list of omics data **</span></span>
<span id="cb2-12" class="hljs-ln-code"><a href="#cb2-12"></a>X <span class="ot">&lt;-</span> <span class="fu">list</span>(X1, X2)</span>
<span id="cb2-13" class="hljs-ln-code"><a href="#cb2-13"></a><span class="co"># number of omics dataset **</span></span>
<span id="cb2-14" class="hljs-ln-code"><a href="#cb2-14"></a>num_omics <span class="ot">&lt;-</span> <span class="dv">2</span></span>
<span id="cb2-15" class="hljs-ln-code"><a href="#cb2-15"></a><span class="co"># tuning parameter candidate length for each omics data</span></span>
<span id="cb2-16" class="hljs-ln-code"><a href="#cb2-16"></a>tuneLength <span class="ot">&lt;-</span> <span class="dv">5</span></span>
<span id="cb2-17" class="hljs-ln-code"><a href="#cb2-17"></a><span class="co"># tuning parameter candadate range for each omics data</span></span>
<span id="cb2-18" class="hljs-ln-code"><a href="#cb2-18"></a>minTune <span class="ot">&lt;-</span> <span class="fl">0.1</span></span>
<span id="cb2-19" class="hljs-ln-code"><a href="#cb2-19"></a>maxTune <span class="ot">&lt;-</span> <span class="fl">0.5</span></span>
<span id="cb2-20" class="hljs-ln-code"><a href="#cb2-20"></a><span class="co"># create empty matrix to store all possible penalty parameters</span></span>
<span id="cb2-21" class="hljs-ln-code"><a href="#cb2-21"></a>penSelect <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="dv">0</span>, <span class="at">nrow =</span> tuneLength, <span class="at">ncol =</span> num_omics)</span>
<span id="cb2-22" class="hljs-ln-code"><a href="#cb2-22"></a><span class="co"># create sparsity penalty options.</span></span>
<span id="cb2-23" class="hljs-ln-code"><a href="#cb2-23"></a><span class="cf">for</span> (Idx <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">ncol</span>(penSelect))</span>
<span id="cb2-24" class="hljs-ln-code"><a href="#cb2-24"></a>{</span>
<span id="cb2-25" class="hljs-ln-code"><a href="#cb2-25"></a>  penSelect[,Idx] <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="at">from =</span> minTune, </span>
<span id="cb2-26" class="hljs-ln-code"><a href="#cb2-26"></a>                         <span class="at">to =</span> maxTune, </span>
<span id="cb2-27" class="hljs-ln-code"><a href="#cb2-27"></a>                         <span class="at">length.out =</span> tuneLength)</span>
<span id="cb2-28" class="hljs-ln-code"><a href="#cb2-28"></a>}</span>
<span id="cb2-29" class="hljs-ln-code"><a href="#cb2-29"></a><span class="co"># expand grid</span></span>
<span id="cb2-30" class="hljs-ln-code"><a href="#cb2-30"></a><span class="co"># convert matrix to list of columns</span></span>
<span id="cb2-31" class="hljs-ln-code"><a href="#cb2-31"></a>list_cols <span class="ot">&lt;-</span> <span class="fu">as.list</span>(<span class="fu">as.data.frame</span>(penSelect))</span>
<span id="cb2-32" class="hljs-ln-code"><a href="#cb2-32"></a><span class="co"># generate all possible combinations</span></span>
<span id="cb2-33" class="hljs-ln-code"><a href="#cb2-33"></a>PenExpand <span class="ot">&lt;-</span> <span class="fu">do.call</span>(expand.grid, list_cols)</span>
<span id="cb2-34" class="hljs-ln-code"><a href="#cb2-34"></a></span>
<span id="cb2-35" class="hljs-ln-code"><a href="#cb2-35"></a><span class="co"># set a CV directory.</span></span>
<span id="cb2-36" class="hljs-ln-code"><a href="#cb2-36"></a>CVDir <span class="ot">&lt;-</span> <span class="st">"Example3foldCV/"</span> </span>
<span id="cb2-37" class="hljs-ln-code"><a href="#cb2-37"></a><span class="fu">dir.create</span>(CVDir)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="efficient-fold-spliting" class="slide level2 smaller scrollable" data-transition="fade" data-auto-animate="true">
<h2 data-id="quarto-animate-title">Efficient fold spliting</h2>
<div class="sourceCode" id="cb3"><pre class="sourceCode numberSource r number-lines code-with-copy" data-id="quarto-animate-code"><code class="sourceCode r hljs"><span id="cb3-1" class="hljs-ln-code"><a href="#cb3-1"></a><span class="fu">set.seed</span>(<span class="dv">12345</span>) <span class="co"># set random seed.</span></span>
<span id="cb3-2" class="hljs-ln-code"><a href="#cb3-2"></a></span>
<span id="cb3-3" class="hljs-ln-code"><a href="#cb3-3"></a><span class="co"># split data into folds</span></span>
<span id="cb3-4" class="hljs-ln-code"><a href="#cb3-4"></a>foldIdx <span class="ot">&lt;-</span> <span class="fu">suppressWarnings</span>(<span class="fu">split</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(X[[<span class="dv">1</span>]]), <span class="fu">sample</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(X[[<span class="dv">1</span>]]), K)))</span>
<span id="cb3-5" class="hljs-ln-code"><a href="#cb3-5"></a>folddata <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(foldIdx), <span class="cf">function</span>(x){</span>
<span id="cb3-6" class="hljs-ln-code"><a href="#cb3-6"></a>    Y <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(Y)</span>
<span id="cb3-7" class="hljs-ln-code"><a href="#cb3-7"></a>    X_train <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb3-8" class="hljs-ln-code"><a href="#cb3-8"></a>    X_test <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb3-9" class="hljs-ln-code"><a href="#cb3-9"></a>    Y_train <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb3-10" class="hljs-ln-code"><a href="#cb3-10"></a>    Y_test <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb3-11" class="hljs-ln-code"><a href="#cb3-11"></a>    <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(X))</span>
<span id="cb3-12" class="hljs-ln-code"><a href="#cb3-12"></a>    {</span>
<span id="cb3-13" class="hljs-ln-code"><a href="#cb3-13"></a>      X_train[[i]] <span class="ot">&lt;-</span> <span class="fu">scale</span>(X[[i]][<span class="sc">-</span>foldIdx[[x]],])</span>
<span id="cb3-14" class="hljs-ln-code"><a href="#cb3-14"></a>      X_test[[i]] <span class="ot">&lt;-</span> <span class="fu">scale</span>(X[[i]][foldIdx[[x]],])</span>
<span id="cb3-15" class="hljs-ln-code"><a href="#cb3-15"></a>    }</span>
<span id="cb3-16" class="hljs-ln-code"><a href="#cb3-16"></a>    Y_train <span class="ot">&lt;-</span> <span class="fu">scale</span>(Y[<span class="sc">-</span>foldIdx[[x]],])</span>
<span id="cb3-17" class="hljs-ln-code"><a href="#cb3-17"></a>    Y_test <span class="ot">&lt;-</span> <span class="fu">scale</span>(Y[foldIdx[[x]],])</span>
<span id="cb3-18" class="hljs-ln-code"><a href="#cb3-18"></a>    <span class="fu">return</span>(<span class="fu">list</span>(<span class="at">X_train =</span> X_train, <span class="at">X_test =</span> X_test,<span class="at">Y_train =</span> Y_train, </span>
<span id="cb3-19" class="hljs-ln-code"><a href="#cb3-19"></a>                <span class="at">Y_test =</span> Y_test))</span>
<span id="cb3-20" class="hljs-ln-code"><a href="#cb3-20"></a>})</span>
<span id="cb3-21" class="hljs-ln-code"><a href="#cb3-21"></a><span class="co"># name each fold of data</span></span>
<span id="cb3-22" class="hljs-ln-code"><a href="#cb3-22"></a><span class="fu">names</span>(folddata) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">'Fold_'</span>, <span class="dv">1</span><span class="sc">:</span>K)</span>
<span id="cb3-23" class="hljs-ln-code"><a href="#cb3-23"></a><span class="co"># saving all preliminary data into local directory for reproducibility purpose</span></span>
<span id="cb3-24" class="hljs-ln-code"><a href="#cb3-24"></a><span class="fu">save</span>(folddata, PenExpand, </span>
<span id="cb3-25" class="hljs-ln-code"><a href="#cb3-25"></a>       <span class="at">file =</span> <span class="fu">paste0</span>(CVDir, <span class="st">"CVData.RData"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="run-k-fold-cv" class="slide level2 smaller scrollable" data-transition="fade" data-auto-animate="true">
<h2 data-id="quarto-animate-title">Run K-fold CV</h2>
<div class="sourceCode" id="cb4"><pre class="sourceCode numberSource r number-lines code-with-copy" data-id="quarto-animate-code"><code class="sourceCode r hljs"><span id="cb4-1" class="hljs-ln-code"><a href="#cb4-1"></a><span class="co"># load cross-validation data</span></span>
<span id="cb4-2" class="hljs-ln-code"><a href="#cb4-2"></a><span class="fu">load</span>(<span class="fu">paste0</span>(CVDir, <span class="st">"CVData.RData"</span>))</span>
<span id="cb4-3" class="hljs-ln-code"><a href="#cb4-3"></a><span class="co"># create an empty list for storing CV result for each fold</span></span>
<span id="cb4-4" class="hljs-ln-code"><a href="#cb4-4"></a>CVResult <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb4-5" class="hljs-ln-code"><a href="#cb4-5"></a><span class="cf">for</span> (CVidx <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>K)</span>
<span id="cb4-6" class="hljs-ln-code"><a href="#cb4-6"></a>{  </span>
<span id="cb4-7" class="hljs-ln-code"><a href="#cb4-7"></a>    <span class="co"># set scaling factor</span></span>
<span id="cb4-8" class="hljs-ln-code"><a href="#cb4-8"></a>    CCcoef <span class="ot">&lt;-</span> scalingFactor</span>
<span id="cb4-9" class="hljs-ln-code"><a href="#cb4-9"></a>    <span class="co"># create empty vector for storing cross-validation result</span></span>
<span id="cb4-10" class="hljs-ln-code"><a href="#cb4-10"></a>    RhoTrain <span class="ot">&lt;-</span> RhoTest <span class="ot">&lt;-</span> DeltaCor <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="dv">0</span>, <span class="fu">nrow</span>(PenExpand))</span>
<span id="cb4-11" class="hljs-ln-code"><a href="#cb4-11"></a>    <span class="cf">for</span>(idx <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(PenExpand))</span>
<span id="cb4-12" class="hljs-ln-code"><a href="#cb4-12"></a>    {</span>
<span id="cb4-13" class="hljs-ln-code"><a href="#cb4-13"></a>      <span class="co"># consider one pair of sparsity penalties at a time.</span></span>
<span id="cb4-14" class="hljs-ln-code"><a href="#cb4-14"></a>      l <span class="ot">&lt;-</span> PenExpand[idx, ]</span>
<span id="cb4-15" class="hljs-ln-code"><a href="#cb4-15"></a>      <span class="co"># run SmCCA on the subsamples (Figure 1, Step II)</span></span>
<span id="cb4-16" class="hljs-ln-code"><a href="#cb4-16"></a>      Ws <span class="ot">&lt;-</span> <span class="fu">getCanWeightsMulti</span>(folddata[[CVidx]][[<span class="st">"X_train"</span>]], </span>
<span id="cb4-17" class="hljs-ln-code"><a href="#cb4-17"></a>                           <span class="at">Trait =</span> <span class="fu">as.matrix</span>(folddata[[CVidx]][[<span class="st">"Y_train"</span>]]), </span>
<span id="cb4-18" class="hljs-ln-code"><a href="#cb4-18"></a>                           <span class="at">Lambda =</span> <span class="fu">as.numeric</span>(l), <span class="at">NoTrait =</span> <span class="cn">FALSE</span>, </span>
<span id="cb4-19" class="hljs-ln-code"><a href="#cb4-19"></a>                           <span class="at">CCcoef =</span> CCcoef)</span>
<span id="cb4-20" class="hljs-ln-code"><a href="#cb4-20"></a>      <span class="co"># evaluate the canonical correlation for training and testing data</span></span>
<span id="cb4-21" class="hljs-ln-code"><a href="#cb4-21"></a>      rho.train <span class="ot">&lt;-</span>  <span class="fu">getCanCorMulti</span>(<span class="at">X =</span> folddata[[CVidx]][[<span class="st">"X_train"</span>]], </span>
<span id="cb4-22" class="hljs-ln-code"><a href="#cb4-22"></a>                                <span class="at">Y =</span> <span class="fu">as.matrix</span>(folddata[[CVidx]][[<span class="st">"Y_train"</span>]]), </span>
<span id="cb4-23" class="hljs-ln-code"><a href="#cb4-23"></a>                                <span class="at">CCWeight =</span> Ws, </span>
<span id="cb4-24" class="hljs-ln-code"><a href="#cb4-24"></a>                                <span class="at">CCcoef =</span> CCcoef)</span>
<span id="cb4-25" class="hljs-ln-code"><a href="#cb4-25"></a>      rho.test <span class="ot">&lt;-</span>  <span class="fu">getCanCorMulti</span>(<span class="at">X =</span> folddata[[CVidx]][[<span class="st">"X_test"</span>]], </span>
<span id="cb4-26" class="hljs-ln-code"><a href="#cb4-26"></a>                                <span class="at">Y =</span> <span class="fu">as.matrix</span>(folddata[[CVidx]][[<span class="st">"Y_test"</span>]]), </span>
<span id="cb4-27" class="hljs-ln-code"><a href="#cb4-27"></a>                                <span class="at">CCWeight =</span> Ws, </span>
<span id="cb4-28" class="hljs-ln-code"><a href="#cb4-28"></a>                                <span class="at">CCcoef =</span> CCcoef)</span>
<span id="cb4-29" class="hljs-ln-code"><a href="#cb4-29"></a></span>
<span id="cb4-30" class="hljs-ln-code"><a href="#cb4-30"></a>      <span class="co"># store cv result  </span></span>
<span id="cb4-31" class="hljs-ln-code"><a href="#cb4-31"></a>      RhoTrain[idx] <span class="ot">&lt;-</span> <span class="fu">round</span>(rho.train, <span class="at">digits =</span> <span class="dv">5</span>)</span>
<span id="cb4-32" class="hljs-ln-code"><a href="#cb4-32"></a>      RhoTest[idx] <span class="ot">&lt;-</span> <span class="fu">round</span>(rho.test, <span class="at">digits =</span> <span class="dv">5</span>)</span>
<span id="cb4-33" class="hljs-ln-code"><a href="#cb4-33"></a>      DeltaCor[idx] <span class="ot">&lt;-</span> <span class="fu">abs</span>(rho.train <span class="sc">-</span> rho.test)</span>
<span id="cb4-34" class="hljs-ln-code"><a href="#cb4-34"></a>    </span>
<span id="cb4-35" class="hljs-ln-code"><a href="#cb4-35"></a>    }</span>
<span id="cb4-36" class="hljs-ln-code"><a href="#cb4-36"></a>    </span>
<span id="cb4-37" class="hljs-ln-code"><a href="#cb4-37"></a>    <span class="co"># record prediction errors for given CV fold and all sparsity penalty </span></span>
<span id="cb4-38" class="hljs-ln-code"><a href="#cb4-38"></a>    <span class="co"># options.</span></span>
<span id="cb4-39" class="hljs-ln-code"><a href="#cb4-39"></a>    CVResult[[CVidx]] <span class="ot">&lt;-</span> <span class="fu">cbind</span>(RhoTrain, RhoTest, DeltaCor)</span>
<span id="cb4-40" class="hljs-ln-code"><a href="#cb4-40"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="step-3-run-smcca-with-pre-selected-penalty-term" class="slide level2 smaller scrollable" data-transition="fade" data-auto-animate="true">
<h2 data-id="quarto-animate-title">Step 3: Run SmCCA with pre-selected penalty term</h2>
<div>
<ul>
<li class="fragment">SmCCNet 1.0.0: Use prediction error.</li>
<li class="fragment">SmCCNet 2.0.0: Use scaled prediction error.</li>
</ul>
</div>
<div class="fragment">
<p><br></p>
<p>The best penalty term should minimized the scaled prediction error:</p>
<span class="math display">\[\begin{equation}
scaledPredErr = \frac{|trainCC - testCC|}{|testCC|},
\end{equation}\]</span>
<p>where <span class="math inline">\(trainCC\)</span> and <span class="math inline">\(testCC\)</span> is defined as the training canonical correlation and testing canonical correlation respectively.</p>
</div>
<div class="fragment">
<div class="sourceCode" id="cb5"><pre class="sourceCode numberSource r number-lines code-with-copy" data-id="quarto-animate-code"><code class="sourceCode r hljs"><span id="cb5-1" class="hljs-ln-code"><a href="#cb5-1"></a><span class="co"># aggregate CV result and select the best penalty term</span></span>
<span id="cb5-2" class="hljs-ln-code"><a href="#cb5-2"></a>AggregatedCVResult <span class="ot">&lt;-</span> <span class="fu">Reduce</span>(<span class="st">"+"</span>, CVResult) <span class="sc">/</span> <span class="fu">length</span>(CVResult)</span>
<span id="cb5-3" class="hljs-ln-code"><a href="#cb5-3"></a><span class="co"># calculate the evaluation metric of interest</span></span>
<span id="cb5-4" class="hljs-ln-code"><a href="#cb5-4"></a>EvalMetric <span class="ot">&lt;-</span> <span class="fu">apply</span>(AggregatedCVResult, <span class="dv">1</span>, <span class="cf">function</span>(x) {x[<span class="dv">3</span>]<span class="sc">/</span><span class="fu">abs</span>(x[<span class="dv">2</span>])})</span>
<span id="cb5-5" class="hljs-ln-code"><a href="#cb5-5"></a><span class="co"># determine the best CV result</span></span>
<span id="cb5-6" class="hljs-ln-code"><a href="#cb5-6"></a>optIdx <span class="ot">&lt;-</span> <span class="fu">which.min</span>(EvalMetric)</span>
<span id="cb5-7" class="hljs-ln-code"><a href="#cb5-7"></a><span class="co"># combine CV evaluation result with penalty candidates </span></span>
<span id="cb5-8" class="hljs-ln-code"><a href="#cb5-8"></a>overallCVInfo <span class="ot">&lt;-</span> <span class="fu">cbind</span>(PenExpand, AggregatedCVResult, <span class="at">scaledPredError =</span> EvalMetric)</span>
<span id="cb5-9" class="hljs-ln-code"><a href="#cb5-9"></a><span class="co"># set column names for penalty terms </span></span>
<span id="cb5-10" class="hljs-ln-code"><a href="#cb5-10"></a><span class="fu">colnames</span>(overallCVInfo)[<span class="dv">1</span><span class="sc">:</span>num_omics] <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">'l'</span>,<span class="dv">1</span><span class="sc">:</span>num_omics)</span>
<span id="cb5-11" class="hljs-ln-code"><a href="#cb5-11"></a><span class="co"># save overall CV result </span></span>
<span id="cb5-12" class="hljs-ln-code"><a href="#cb5-12"></a><span class="fu">write.csv</span>(overallCVInfo, <span class="at">file =</span> <span class="fu">paste0</span>(CVDir, <span class="st">'overallCVInfo.csv'</span>), </span>
<span id="cb5-13" class="hljs-ln-code"><a href="#cb5-13"></a>          <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb5-14" class="hljs-ln-code"><a href="#cb5-14"></a></span>
<span id="cb5-15" class="hljs-ln-code"><a href="#cb5-15"></a><span class="co"># feature sampling proportions, 0.9 for miRNA since it has less features. **</span></span>
<span id="cb5-16" class="hljs-ln-code"><a href="#cb5-16"></a>s <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fl">0.7</span>,<span class="fl">0.9</span>) </span>
<span id="cb5-17" class="hljs-ln-code"><a href="#cb5-17"></a><span class="co"># number of subsamples.</span></span>
<span id="cb5-18" class="hljs-ln-code"><a href="#cb5-18"></a>SubsamplingNum <span class="ot">&lt;-</span> <span class="dv">50</span></span>
<span id="cb5-19" class="hljs-ln-code"><a href="#cb5-19"></a><span class="co"># run SmCCA on the subsamples (Figure 1, Step II)</span></span>
<span id="cb5-20" class="hljs-ln-code"><a href="#cb5-20"></a>Ws <span class="ot">&lt;-</span> <span class="fu">getRobustWeightsMulti</span>(X, </span>
<span id="cb5-21" class="hljs-ln-code"><a href="#cb5-21"></a>                        <span class="at">Trait =</span> <span class="fu">as.matrix</span>(Y), </span>
<span id="cb5-22" class="hljs-ln-code"><a href="#cb5-22"></a>                        <span class="at">NoTrait =</span> <span class="cn">FALSE</span>,<span class="at">CCcoef =</span> scalingFactor,</span>
<span id="cb5-23" class="hljs-ln-code"><a href="#cb5-23"></a>                        <span class="at">Lambda =</span> <span class="fu">as.numeric</span>(overallCVInfo[optIdx,<span class="dv">1</span><span class="sc">:</span>num_omics]),</span>
<span id="cb5-24" class="hljs-ln-code"><a href="#cb5-24"></a>                        <span class="at">s =</span> s, </span>
<span id="cb5-25" class="hljs-ln-code"><a href="#cb5-25"></a>                        <span class="at">SubsamplingNum =</span> SubsamplingNum)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="step-4-network-clustering" class="slide level2 smaller scrollable" data-transition="fade" data-auto-animate="true">
<h2 data-id="quarto-animate-title">Step 4: Network clustering</h2>
<p>Get multi-omics modules by constructing global adjacency matrix and applying hierarchical tree cutting and plotting the reconstructed networks:</p>
<div class="fragment">
<div class="sourceCode" id="cb6"><pre class="sourceCode numberSource r number-lines code-with-copy" data-id="quarto-animate-code"><code class="sourceCode r hljs"><span id="cb6-1" class="hljs-ln-code"><a href="#cb6-1"></a><span class="co"># construct adjacency matrix</span></span>
<span id="cb6-2" class="hljs-ln-code"><a href="#cb6-2"></a>Abar <span class="ot">&lt;-</span> <span class="fu">getAbar</span>(Ws, <span class="at">FeatureLabel =</span> AbarLabel)</span>
<span id="cb6-3" class="hljs-ln-code"><a href="#cb6-3"></a><span class="co"># perform clustering based on the adjacency matrix Abar </span></span>
<span id="cb6-4" class="hljs-ln-code"><a href="#cb6-4"></a>OmicsModule <span class="ot">&lt;-</span> <span class="fu">getOmicsModules</span>(Abar, <span class="at">PlotTree =</span> <span class="cn">FALSE</span>)</span>
<span id="cb6-5" class="hljs-ln-code"><a href="#cb6-5"></a><span class="fu">save</span>(Ws, Abar, OmicsModule, <span class="at">file =</span> <span class="fu">paste0</span>(CVDir, <span class="st">"SmCCNetWeights.RData"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="step-5-network-pruning-algorithm-new-feature" class="slide level2 smaller scrollable" data-transition="fade" data-auto-animate="true">
<h2 data-id="quarto-animate-title">Step 5: Network pruning algorithm (new feature)</h2>
<div>
<ul>
<li class="fragment">SmCCNet 1.0.0: No network pruning algorithm.</li>
<li class="fragment">SmCCNet 2.0.0: Novel network pruning algorithm.</li>
<li class="fragment">Unnecessary features (nodes) will be filtered from the original network module.</li>
<li class="fragment">2 key parameters:
<ul>
<li class="fragment">baseline network size (e.g.&nbsp;10)</li>
<li class="fragment">maximum possible network size (e.g.&nbsp;100)</li>
</ul></li>
<li class="fragment">2 objectives:
<ul>
<li class="fragment">network highly correlated from phenotype</li>
<li class="fragment">network highly correlated with baseline network</li>
</ul></li>
</ul>
</div>
<div class="fragment">
<div class="sourceCode" id="cb7"><pre class="sourceCode numberSource r number-lines code-with-copy" data-id="quarto-animate-code"><code class="sourceCode r hljs"><span id="cb7-1" class="hljs-ln-code"><a href="#cb7-1"></a><span class="co"># make sure there are no duplicated labels</span></span>
<span id="cb7-2" class="hljs-ln-code"><a href="#cb7-2"></a>AbarLabel <span class="ot">&lt;-</span> <span class="fu">make.unique</span>(AbarLabel)</span>
<span id="cb7-3" class="hljs-ln-code"><a href="#cb7-3"></a></span>
<span id="cb7-4" class="hljs-ln-code"><a href="#cb7-4"></a><span class="co"># create concatenate omics data for network pruning</span></span>
<span id="cb7-5" class="hljs-ln-code"><a href="#cb7-5"></a>X_big <span class="ot">&lt;-</span> <span class="fu">cbind</span>(X1,X2)</span>
<span id="cb7-6" class="hljs-ln-code"><a href="#cb7-6"></a></span>
<span id="cb7-7" class="hljs-ln-code"><a href="#cb7-7"></a><span class="co"># calculate feature correlation matrix</span></span>
<span id="cb7-8" class="hljs-ln-code"><a href="#cb7-8"></a>bigCor2 <span class="ot">&lt;-</span> <span class="fu">cor</span>(X_big)</span>
<span id="cb7-9" class="hljs-ln-code"><a href="#cb7-9"></a></span>
<span id="cb7-10" class="hljs-ln-code"><a href="#cb7-10"></a><span class="co"># data type</span></span>
<span id="cb7-11" class="hljs-ln-code"><a href="#cb7-11"></a>types <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">'gene'</span>, <span class="fu">ncol</span>(X1)), <span class="fu">rep</span>(<span class="st">'mirna'</span>, <span class="fu">ncol</span>(X2)))</span>
<span id="cb7-12" class="hljs-ln-code"><a href="#cb7-12"></a></span>
<span id="cb7-13" class="hljs-ln-code"><a href="#cb7-13"></a><span class="co"># filter out network modules with insufficient number of nodes</span></span>
<span id="cb7-14" class="hljs-ln-code"><a href="#cb7-14"></a>module_length <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">lapply</span>(OmicsModule, length))</span>
<span id="cb7-15" class="hljs-ln-code"><a href="#cb7-15"></a>network_modules <span class="ot">&lt;-</span> OmicsModule[module_length <span class="sc">&gt;</span> <span class="dv">10</span>]</span>
<span id="cb7-16" class="hljs-ln-code"><a href="#cb7-16"></a><span class="co"># extract pruned network modules</span></span>
<span id="cb7-17" class="hljs-ln-code"><a href="#cb7-17"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(network_modules))</span>
<span id="cb7-18" class="hljs-ln-code"><a href="#cb7-18"></a>{</span>
<span id="cb7-19" class="hljs-ln-code"><a href="#cb7-19"></a>  <span class="fu">cat</span>(<span class="fu">paste0</span>(<span class="st">'For network module: '</span>, i, <span class="st">'</span><span class="sc">\n</span><span class="st">'</span>))</span>
<span id="cb7-20" class="hljs-ln-code"><a href="#cb7-20"></a>  <span class="co"># define subnetwork</span></span>
<span id="cb7-21" class="hljs-ln-code"><a href="#cb7-21"></a>  abar_sub <span class="ot">&lt;-</span> Abar[network_modules[[i]],network_modules[[i]]]</span>
<span id="cb7-22" class="hljs-ln-code"><a href="#cb7-22"></a>  cor_sub <span class="ot">&lt;-</span> bigCor2[network_modules[[i]],network_modules[[i]]]</span>
<span id="cb7-23" class="hljs-ln-code"><a href="#cb7-23"></a>  <span class="co"># prune network module</span></span>
<span id="cb7-24" class="hljs-ln-code"><a href="#cb7-24"></a>  <span class="fu">networkPruning</span>(<span class="at">Abar =</span> abar_sub,<span class="at">CorrMatrix =</span> cor_sub, </span>
<span id="cb7-25" class="hljs-ln-code"><a href="#cb7-25"></a>                          <span class="at">type =</span> types[network_modules[[i]]], </span>
<span id="cb7-26" class="hljs-ln-code"><a href="#cb7-26"></a>                 <span class="at">data =</span> X_big[,network_modules[[i]]],      </span>
<span id="cb7-27" class="hljs-ln-code"><a href="#cb7-27"></a>              <span class="at">Pheno =</span> Y, <span class="at">ModuleIdx =</span> i, <span class="at">min_mod_size =</span> <span class="dv">10</span>, </span>
<span id="cb7-28" class="hljs-ln-code"><a href="#cb7-28"></a>                          <span class="at">max_mod_size =</span> <span class="dv">100</span>, <span class="at">method =</span> <span class="st">'NetSHy'</span>, </span>
<span id="cb7-29" class="hljs-ln-code"><a href="#cb7-29"></a>                          <span class="at">saving_dir =</span> CVDir)</span>
<span id="cb7-30" class="hljs-ln-code"><a href="#cb7-30"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="step-6-network-visualization" class="slide level2 smaller scrollable" data-transition="fade" data-auto-animate="true">
<h2 data-id="quarto-animate-title">Step 6: Network visualization</h2>
<div>
<ul>
<li class="fragment">SmCCNet 1.0.0: Use igraph to visualize network.</li>
<li class="fragment">SmCCNet 2.0.0: Use cytoscape software to visualize network.
<ul>
<li class="fragment">Use correlation matrix to filter out weak edges.</li>
<li class="fragment">Need to install cytoscape and open it first.</li>
<li class="fragment">Use RCy3 package to communicate between R and cytoscape.</li>
</ul></li>
<li class="fragment">SmCCNet 2.0.0: Use ShinyApp to visualize network.
<ul>
<li class="fragment">https://smccnet.shinyapps.io/smccnetnetwork/</li>
</ul></li>
</ul>
</div>
<div class="fragment">
<p><br></p>
<p><img data-src="images/MultiOmicsNet.png"></p>
</div>
</section>
<section id="visualization-with-cytoscape" class="slide level2" data-background-video="images/cytoscape.mp4">
<h2>Visualization with cytoscape</h2>
</section>
<section id="method-of-scaling-factors-selection-new-feature" class="slide level2 smaller scrollable" data-transition="fade" data-auto-animate="true">
<h2 data-id="quarto-animate-title">Method of scaling factors selection (new feature)</h2>
<div>
<ul>
<li class="fragment">SmCCNet 1.0.0: User-defined (two omics only)</li>
<li class="fragment">SmCCNet 2.0.0: 3 new methods (for more than two omics):
<ul>
<li class="fragment">Use prompt to define</li>
<li class="fragment">Selection based on pairwise correlation</li>
<li class="fragment">Selection based on cross-validation</li>
</ul></li>
</ul>
</div>
</section>
<section id="method-1-use-prompt-to-define" class="slide level2" data-background-video="images/scalingFactor.mp4">
<h2>Method 1: Use Prompt to define</h2>
</section>
<section id="method-2-selection-based-on-pairwise-correlation" class="slide level2 smaller scrollable" data-transition="fade" data-auto-animate="true">
<h2 data-id="quarto-animate-title">Method 2: Selection based on pairwise correlation</h2>
<div>
<ul>
<li class="fragment">Employ sparse canonical correlation analysis (SCCA) for pairwise omics data correlation.</li>
<li class="fragment">Use stringent penalty pair (<span class="math inline">\(0.1,0.1\)</span>) for canonical correlation.</li>
<li class="fragment">Canonical correlation value is used as between-omics scaling factor; 1 for omics-phenotype.</li>
<li class="fragment">Introduce shrinkage factor to prioritize omics-phenotype relationships.</li>
<li class="fragment">Two-Omics Example: With omics-omics correlation of 0.8 and shrinkage of 2, scaling factors are <span class="math inline">\((a,b_1,b_2) = c(0.4, 1, 1)\)</span>.</li>
<li class="fragment">Implemented in <strong>fastAutoSmCCNet()</strong>; refer to automated SmCCNet vignette.</li>
</ul>
</div>
</section>
<section id="method-3-use-cross-validation-to-select-scaling-factors" class="slide level2 smaller scrollable" data-transition="fade" data-auto-animate="true">
<h2 data-id="quarto-animate-title">Method 3: Use cross-validation to select scaling factors</h2>
<div>
<ul>
<li class="fragment"><p>Two omics data: mRNA and miRNA with a quantitative phenotype.</p></li>
<li class="fragment"><p>Goal: Evaluate performance of scaling factors for each scaling factors combination.</p></li>
<li class="fragment"><p>Create candidate scaling factors:</p>
<ul>
<li class="fragment"><p>Set all omics-omics scaling factor to 1.</p></li>
<li class="fragment"><p>Vary the omics-phenotype scaling factors.</p></li>
</ul></li>
<li class="fragment"><p>Ensure scaling factors sum up to 1 (e.g., <span class="math inline">\(a_{1,2} + b_1 + b_2 = 1\)</span>).</p></li>
<li class="fragment"><p>Use nested grid search to select optimal scaling factors and penalty parameters.</p></li>
<li class="fragment"><p>For each scaling factor combination:</p>
<ul>
<li class="fragment"><p>Choose sparse penalty parameters (l1, l2) via K-fold cross validation.</p></li>
<li class="fragment"><p>Minimize scaled prediction error to determine best penalty pair.</p></li>
</ul></li>
<li class="fragment"><p>Record best scaled prediction error for each scaling factor set to find optimal factors.</p></li>
<li class="fragment"><p>Refer to SmCCNet multi-omics vignette for code and implementation.</p></li>
</ul>
</div>
</section>
<section id="multi-omics-with-binary-phenotype-new-feature" class="slide level2 smaller scrollable" data-transition="fade" data-auto-animate="true">
<h2 data-id="quarto-animate-title">Multi-omics with binary phenotype (new feature)</h2>
<div>
<ul>
<li class="fragment"><p>Hybrid step-wise approach incorporates:</p>
<ul>
<li class="fragment">Sparse Multiple Canonical Correlation Analysis (SmCCA)</li>
<li class="fragment">Sparse Partial Least Squared Discriminant Analysis (SPLSDA)</li>
</ul></li>
<li class="fragment"><p>Step 1: SmCCA on Omics data</p>
<ul>
<li class="fragment"><p>Objective: find molecular features with high connectivity.</p></li>
<li class="fragment"><p>Use relaxed penalty parameter to include as many molecular features as possible.</p></li>
<li class="fragment"><p>Select molecular features for step 2.</p></li>
<li class="fragment"><p>Obtain feature weight vectors.</p></li>
</ul></li>
<li class="fragment"><p>Step 2: SPLSDA on phenotype</p>
<ul>
<li class="fragment"><p>Concatenate selected molecular features</p></li>
<li class="fragment"><p>Run SPLSDA</p></li>
<li class="fragment"><p>Obtain feature weight vectors.</p></li>
</ul></li>
<li class="fragment"><p>Final feature weight for global network construction:</p></li>
</ul>
<pre><code>-   Weighted sum of SmCCA feature weight vectors and SPLSDA feature weight vectors.</code></pre>
</div>
<div class="fragment">
<p><br> Note: when selecting the best penalty term with performance measure, only run through step 1-5 to obtain prediction performance. For more detail, please refer to the multi-omics vignette.</p>
</div>
</section>
<section id="prediction-performance-measures-new-feature" class="slide level2 smaller scrollable" data-transition="fade" data-auto-animate="true">
<h2 data-id="quarto-animate-title">Prediction performance measures (new feature)</h2>
<div>
<ul>
<li class="fragment">Prediction Accuracy: Ideal when both disease statuses are evenly distributed. Measures overall correct predictions.</li>
<li class="fragment">AUC Score: Assesses model’s ability to distinguish between disease statuses across varying thresholds.</li>
<li class="fragment">Precision: Prioritized when falsely labeling a disease-negative as positive is costly. Measures true positives among predicted positives.</li>
<li class="fragment">Recall: Essential when detecting as many disease-positives is crucial, even if some disease-negatives are misclassified.</li>
<li class="fragment">F1 Score: Harmonic mean of precision and recall; best for uneven class distributions.</li>
</ul>
</div>
</section>
<section id="single-omics-smccnet-new-feature" class="slide level2 smaller scrollable" data-transition="fade" data-auto-animate="true">
<h2 data-id="quarto-animate-title">Single-omics SmCCNet (new feature)</h2>
<div>
<ul>
<li class="fragment">SmCCNet 1.0.0: can’t work with single-omics data.</li>
<li class="fragment">SmCCNet 2.0.0: work with single-omics data with both quantitative and binary phenotype.</li>
<li class="fragment">Single-omics with quantitative phenotype: no scaling factor.</li>
<li class="fragment">Single-omics with binary phenotype: don’t have to run SmCCA.</li>
<li class="fragment">Refer to single-omics SmCCNet vignette for more detail.</li>
</ul>
</div>
<div class="fragment">
<p><br></p>
</div>
</section>
<section id="fast-automated-smccnet-new-feature" class="slide level2 smaller scrollable" data-transition="fade" data-auto-animate="true">
<h2 data-id="quarto-animate-title">Fast Automated SmCCNet (new feature)</h2>
<div>
<ul>
<li class="fragment">SmCCNet 1.0.0: set up pipeline with 200-500 lines of code.</li>
<li class="fragment">SmCCNet 2.0.0: end-to-end pipeline with 1 line of code.</li>
<li class="fragment">Automatically detect single vs.&nbsp;multi-omics, binary vs.&nbsp;quantitative phenotype.</li>
<li class="fragment">Automatically select scaling factors and subsampling percentage.</li>
<li class="fragment">Speed up cross-validation with parallel computing.</li>
<li class="fragment">Organized progress output in console.</li>
<li class="fragment">Supply only the omics and phenotype data, output the subnetworks (stored in local directory).</li>
<li class="fragment">Save cross-validation result and cross-validation data in user-specified directory for reproducibility.</li>
</ul>
</div>
<div class="fragment">
<p><br></p>
</div>
</section>
<section id="fast-automated-smccnet-arguments" class="slide level2 smaller scrollable" data-transition="fade" data-auto-animate="true">
<h2 data-id="quarto-animate-title">Fast Automated SmCCNet Arguments</h2>
<table>
<colgroup>
<col style="width: 22%">
<col style="width: 77%">
</colgroup>
<thead>
<tr class="header">
<th>Argument</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><strong>X</strong></td>
<td>A list of omics matrices with same set and order of subjects</td>
</tr>
<tr class="even">
<td><strong>Y</strong></td>
<td>Phenotype variable of either numeric or binary, for binary variable, for binary Y, it should be binarized to 0 and 1 before running this function.</td>
</tr>
<tr class="odd">
<td><strong>AdjustedCovar</strong></td>
<td>A data frame of covariates of interest to be adjusted for through regressing-out approach, <strong>preprocess</strong> must be set to TRUE.</td>
</tr>
<tr class="even">
<td><strong>Kfold</strong></td>
<td>Number of folds for cross-validation, default is set to 5.</td>
</tr>
<tr class="odd">
<td><strong>EvalMethod</strong></td>
<td>For single or multi-omics with binary phenotype, the evaluation methods used to selected the optimal penalty parameter(s). The selections is among ‘accuracy’, ‘auc’, ‘precision’, ‘recall’, and ‘f1’, default is set to ‘accuracy’.</td>
</tr>
<tr class="even">
<td><strong>subSampNum</strong></td>
<td>Number of subsampling to run, the higher the better in terms of accuracy, but at a cost of computational time, we generally recommend 500-1000 to increase robustness.</td>
</tr>
<tr class="odd">
<td><strong>BetweenShrinkage</strong></td>
<td>A real number &gt; 0 that helps shrink the importance of omics-omics correlation component, the larger this number is, the greater the shrinkage it is, default is set to 2.</td>
</tr>
<tr class="even">
<td><strong>ScalingPen</strong></td>
<td>A numeric vector of length 2 used as the penalty terms for scaling factor selection method, default set to 0.1, and should be between 0 and 1.</td>
</tr>
<tr class="odd">
<td><strong>DataType</strong></td>
<td>A vector indicating omics type for each element of X, example would be c(‘gene’, ‘miRNA’).</td>
</tr>
<tr class="even">
<td><strong>CutHeight</strong></td>
<td>A numeric value specifying the cut height for hierarchical clustering, should be between 0 and 1, default is set to <span class="math inline">\(1-0.1^{10}\)</span>.</td>
</tr>
<tr class="odd">
<td><strong>min_size</strong></td>
<td>Minimally possible subnetwork size after network pruning, default set to 10.</td>
</tr>
<tr class="even">
<td><strong>max_size</strong></td>
<td>Maximally possible subnetwork size after network pruning, default set to 100.</td>
</tr>
<tr class="odd">
<td><strong>summarization</strong></td>
<td>Summarization method used for network pruning and summarization, should be either ‘NetSHy’ or ‘PCA’, default is set to ‘NetSHy’.</td>
</tr>
<tr class="even">
<td><strong>saving_dir</strong></td>
<td>Directory where user would like to store the subnetwork results, default is set to the current working directory.</td>
</tr>
<tr class="odd">
<td><strong>preprocess</strong></td>
<td>TRUE or FALSE, Whether the data preprocessing step should be conducted, default is set to FALSE. If covariates adjustment is needed, add covariates to the <strong>AdjustedCovar</strong> argument.</td>
</tr>
<tr class="even">
<td><strong>ncomp_pls</strong></td>
<td>Number of components for PLS algorithm, only used when binary phenotype is given, default is set to 3.</td>
</tr>
<tr class="odd">
<td><strong>tuneLength</strong></td>
<td>The total number of candidate penalty term values for each omics data, default is set to 5.</td>
</tr>
<tr class="even">
<td><strong>tuneRangeCCA</strong></td>
<td>A vector of length 2 that represents the range of candidate penalty term values for each omics data based on canonical correlation analysis, default is set to c(0.1,0.5).</td>
</tr>
<tr class="odd">
<td><strong>tuneRangePLS</strong></td>
<td>A vector of length 2 that represents the range of candidate penalty term values for each omics data based on partial least squared discriminant analysis, default is set to c(0.5,0.9).</td>
</tr>
<tr class="even">
<td><strong>seed</strong></td>
<td>Random seed for result reproducibility, default is set to 123.</td>
</tr>
</tbody>
</table>
</section>
<section id="code-example" class="slide level2 smaller scrollable" data-transition="fade" data-auto-animate="true">
<h2 data-id="quarto-animate-title">Code example</h2>
<div class="sourceCode" id="cb9" data-code-line-numbers="1-5|6-10|11-15|16-23|24-30|1-30"><pre class="sourceCode numberSource r number-lines code-with-copy" data-id="quarto-animate-code"><code class="sourceCode r hljs"><span id="cb9-1" class="hljs-ln-code"><a href="#cb9-1"></a><span class="fu">library</span>(SmCCNet)</span>
<span id="cb9-2" class="hljs-ln-code"><a href="#cb9-2"></a><span class="fu">set.seed</span>(<span class="dv">123</span>)</span>
<span id="cb9-3" class="hljs-ln-code"><a href="#cb9-3"></a><span class="fu">data</span>(<span class="st">"ExampleData"</span>)</span>
<span id="cb9-4" class="hljs-ln-code"><a href="#cb9-4"></a>Y_binary <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(Y <span class="sc">&gt;</span> <span class="fu">quantile</span>(Y, <span class="fl">0.5</span>), <span class="dv">1</span>, <span class="dv">0</span>)</span>
<span id="cb9-5" class="hljs-ln-code"><a href="#cb9-5"></a><span class="co"># single-omics PLS</span></span>
<span id="cb9-6" class="hljs-ln-code"><a href="#cb9-6"></a>result <span class="ot">&lt;-</span> <span class="fu">fastAutoSmCCNet</span>(<span class="at">X =</span> <span class="fu">list</span>(X1), <span class="at">Y =</span> <span class="fu">as.factor</span>(Y_binary), <span class="at">Kfold =</span> <span class="dv">3</span>, </span>
<span id="cb9-7" class="hljs-ln-code"><a href="#cb9-7"></a>                          <span class="at">subSampNum =</span> <span class="dv">100</span>, <span class="at">DataType =</span> <span class="fu">c</span>(<span class="st">'Gene'</span>),</span>
<span id="cb9-8" class="hljs-ln-code"><a href="#cb9-8"></a>                          <span class="at">saving_dir =</span> <span class="fu">getwd</span>(), <span class="at">EvalMethod =</span> <span class="st">'auc'</span>, </span>
<span id="cb9-9" class="hljs-ln-code"><a href="#cb9-9"></a>                          <span class="at">summarization =</span> <span class="st">'NetSHy'</span>, </span>
<span id="cb9-10" class="hljs-ln-code"><a href="#cb9-10"></a>                          <span class="at">CutHeight =</span> <span class="dv">1</span> <span class="sc">-</span> <span class="fl">0.1</span><span class="sc">^</span><span class="dv">10</span>, <span class="at">ncomp_pls =</span> <span class="dv">5</span>)</span>
<span id="cb9-11" class="hljs-ln-code"><a href="#cb9-11"></a><span class="co"># single-omics CCA</span></span>
<span id="cb9-12" class="hljs-ln-code"><a href="#cb9-12"></a>result <span class="ot">&lt;-</span> <span class="fu">fastAutoSmCCNet</span>(<span class="at">X =</span> <span class="fu">list</span>(X1), <span class="at">Y =</span> Y, <span class="at">Kfold =</span> <span class="dv">3</span>, <span class="at">preprocess =</span> <span class="cn">FALSE</span>,</span>
<span id="cb9-13" class="hljs-ln-code"><a href="#cb9-13"></a>                          <span class="at">subSampNum =</span> <span class="dv">50</span>, <span class="at">DataType =</span> <span class="fu">c</span>(<span class="st">'Gene'</span>),</span>
<span id="cb9-14" class="hljs-ln-code"><a href="#cb9-14"></a>                          <span class="at">saving_dir =</span> <span class="fu">getwd</span>(), <span class="at">summarization =</span> <span class="st">'NetSHy'</span>,</span>
<span id="cb9-15" class="hljs-ln-code"><a href="#cb9-15"></a>                          <span class="at">CutHeight =</span> <span class="dv">1</span> <span class="sc">-</span> <span class="fl">0.1</span><span class="sc">^</span><span class="dv">10</span>)</span>
<span id="cb9-16" class="hljs-ln-code"><a href="#cb9-16"></a><span class="co"># multi-omics PLS</span></span>
<span id="cb9-17" class="hljs-ln-code"><a href="#cb9-17"></a>result <span class="ot">&lt;-</span> <span class="fu">fastAutoSmCCNet</span>(<span class="at">X =</span> <span class="fu">list</span>(X1,X2), <span class="at">Y =</span> <span class="fu">as.factor</span>(Y_binary), </span>
<span id="cb9-18" class="hljs-ln-code"><a href="#cb9-18"></a>                          <span class="at">Kfold =</span> <span class="dv">3</span>, <span class="at">subSampNum =</span> <span class="dv">50</span>, </span>
<span id="cb9-19" class="hljs-ln-code"><a href="#cb9-19"></a>                          <span class="at">DataType =</span> <span class="fu">c</span>(<span class="st">'Gene'</span>, <span class="st">'miRNA'</span>), </span>
<span id="cb9-20" class="hljs-ln-code"><a href="#cb9-20"></a>                          <span class="at">CutHeight =</span> <span class="dv">1</span> <span class="sc">-</span> <span class="fl">0.1</span><span class="sc">^</span><span class="dv">10</span>,</span>
<span id="cb9-21" class="hljs-ln-code"><a href="#cb9-21"></a>                          <span class="at">saving_dir =</span> <span class="fu">getwd</span>(), <span class="at">EvalMethod =</span> <span class="st">'auc'</span>, </span>
<span id="cb9-22" class="hljs-ln-code"><a href="#cb9-22"></a>                          <span class="at">summarization =</span> <span class="st">'NetSHy'</span>,</span>
<span id="cb9-23" class="hljs-ln-code"><a href="#cb9-23"></a>                          <span class="at">BetweenShrinkage =</span> <span class="dv">5</span>, <span class="at">ncomp_pls =</span> <span class="dv">3</span>)</span>
<span id="cb9-24" class="hljs-ln-code"><a href="#cb9-24"></a><span class="co"># multi-omics CCA</span></span>
<span id="cb9-25" class="hljs-ln-code"><a href="#cb9-25"></a>result <span class="ot">&lt;-</span> <span class="fu">fastAutoSmCCNet</span>(<span class="at">X =</span> <span class="fu">list</span>(X1,X2), <span class="at">Y =</span> Y, </span>
<span id="cb9-26" class="hljs-ln-code"><a href="#cb9-26"></a>                          <span class="at">K =</span> <span class="dv">3</span>, <span class="at">subSampNum =</span> <span class="dv">50</span>, <span class="at">DataType =</span> <span class="fu">c</span>(<span class="st">'Gene'</span>, <span class="st">'miRNA'</span>), </span>
<span id="cb9-27" class="hljs-ln-code"><a href="#cb9-27"></a>                          <span class="at">CutHeight =</span> <span class="dv">1</span> <span class="sc">-</span> <span class="fl">0.1</span><span class="sc">^</span><span class="dv">10</span>,</span>
<span id="cb9-28" class="hljs-ln-code"><a href="#cb9-28"></a>                          <span class="at">saving_dir =</span> <span class="fu">getwd</span>(),  </span>
<span id="cb9-29" class="hljs-ln-code"><a href="#cb9-29"></a>                          <span class="at">summarization =</span> <span class="st">'NetSHy'</span>,</span>
<span id="cb9-30" class="hljs-ln-code"><a href="#cb9-30"></a>                          <span class="at">BetweenShrinkage =</span> <span class="dv">5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="demo" class="slide level2" data-background-video="images/fastAutoSmCCNet.mp4">
<h2>Demo</h2>
</section>
<section id="summary" class="slide level2 smaller scrollable" data-transition="fade" data-auto-animate="true">
<h2 data-id="quarto-animate-title">Summary</h2>
<div>
<ul>
<li class="fragment">Second-generation SmCCNet is generalizable to more than two omics data (with subtle adjustment on a few line of code).</li>
<li class="fragment">Single-omics analysis is enabled.</li>
<li class="fragment">Novel statistical method is developed.</li>
<li class="fragment">Less hard-coded content.</li>
<li class="fragment">Less output saved in local directory.</li>
<li class="fragment">New features are available (scaling factors, pruning, and more…).</li>
</ul>
</div>
<div class="fragment">
<p><br></p>
</div>
</section>
<section>
<section id="availability" class="title-slide slide level1 smaller scrollable center" data-transition="fade" data-auto-animate="true">
<h1 data-id="quarto-animate-title">Availability</h1>
<div>
<ul>
<li class="fragment">CRAN: https://cran.r-project.org/web/packages/SmCCNet/index.html</li>
<li class="fragment">GitHub: https://github.com/KechrisLab/SmCCNet</li>
<li class="fragment">Network Visualization Tool: https://smccnet.shinyapps.io/smccnetnetwork/</li>
</ul>
</div>
</section>
<section id="acknowledgement" class="slide level2 smaller scrollable" data-transition="fade" data-auto-animate="true">
<h2 data-id="quarto-animate-title">Acknowledgement</h2>
<p>We would like to acknowledge the original contributors to the SmCCNet R package, including Yonghua Zhuang, W. Jenny Shi, and Laura Saba. Their collective contributions to the earlier versions of SmCCNet have been instrumental in shaping its development.</p>
<p>We are also deeply grateful to Dr.Katerina Kechris, for her continuous guidance and support, which has been vital in the refinement of this package.</p>
<p>Special thanks to Thao Vu, Iain Konigsberg, and Katherine Pratte for their invaluable feedback and internal testing of the package.</p>
<p>The SmCCNet R package owes much of its growth and functionality to these original contributors, and we appreciate their enduring influence on its development.</p>

<img src="images/logo.png" class="slide-logo r-stretch"><div class="footer footer-default">
<p><a href="https://quarto.org" class="uri">https://quarto.org</a></p>
</div>
</section></section>
    </div>
  </div>

  <script>window.backupDefine = window.define; window.define = undefined;</script>
  <script src="quarto_presentation_files/libs/revealjs/dist/reveal.js"></script>
  <!-- reveal.js plugins -->
  <script src="quarto_presentation_files/libs/revealjs/plugin/quarto-line-highlight/line-highlight.js"></script>
  <script src="quarto_presentation_files/libs/revealjs/plugin/pdf-export/pdfexport.js"></script>
  <script src="quarto_presentation_files/libs/revealjs/plugin/reveal-menu/menu.js"></script>
  <script src="quarto_presentation_files/libs/revealjs/plugin/reveal-menu/quarto-menu.js"></script>
  <script src="quarto_presentation_files/libs/revealjs/plugin/reveal-chalkboard/plugin.js"></script>
  <script src="quarto_presentation_files/libs/revealjs/plugin/quarto-support/support.js"></script>
  

  <script src="quarto_presentation_files/libs/revealjs/plugin/notes/notes.js"></script>
  <script src="quarto_presentation_files/libs/revealjs/plugin/search/search.js"></script>
  <script src="quarto_presentation_files/libs/revealjs/plugin/zoom/zoom.js"></script>
  <script src="quarto_presentation_files/libs/revealjs/plugin/math/math.js"></script>
  <script>window.define = window.backupDefine; window.backupDefine = undefined;</script>

  <script>

      // Full list of configuration options available at:
      // https://revealjs.com/config/
      Reveal.initialize({
'controlsAuto': true,
'previewLinksAuto': true,
'smaller': false,
'pdfSeparateFragments': false,
'autoAnimateEasing': "ease",
'autoAnimateDuration': 1,
'autoAnimateUnmatched': true,
'menu': {"side":"left","useTextContentForMissingTitles":true,"markers":false,"loadIcons":false,"custom":[{"title":"Tools","icon":"<i class=\"fas fa-gear\"></i>","content":"<ul class=\"slide-menu-items\">\n<li class=\"slide-tool-item active\" data-item=\"0\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.fullscreen(event)\"><kbd>f</kbd> Fullscreen</a></li>\n<li class=\"slide-tool-item\" data-item=\"1\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.speakerMode(event)\"><kbd>s</kbd> Speaker View</a></li>\n<li class=\"slide-tool-item\" data-item=\"2\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.overview(event)\"><kbd>o</kbd> Slide Overview</a></li>\n<li class=\"slide-tool-item\" data-item=\"3\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.togglePdfExport(event)\"><kbd>e</kbd> PDF Export Mode</a></li>\n<li class=\"slide-tool-item\" data-item=\"4\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.toggleChalkboard(event)\"><kbd>b</kbd> Toggle Chalkboard</a></li>\n<li class=\"slide-tool-item\" data-item=\"5\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.toggleNotesCanvas(event)\"><kbd>c</kbd> Toggle Notes Canvas</a></li>\n<li class=\"slide-tool-item\" data-item=\"6\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.downloadDrawings(event)\"><kbd>d</kbd> Download Drawings</a></li>\n<li class=\"slide-tool-item\" data-item=\"7\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.keyboardHelp(event)\"><kbd>?</kbd> Keyboard Help</a></li>\n</ul>"}],"openButton":true},
'chalkboard': {"buttons":true},
'smaller': false,
 
        // Display controls in the bottom right corner
        controls: false,

        // Help the user learn the controls by providing hints, for example by
        // bouncing the down arrow when they first encounter a vertical slide
        controlsTutorial: false,

        // Determines where controls appear, "edges" or "bottom-right"
        controlsLayout: 'edges',

        // Visibility rule for backwards navigation arrows; "faded", "hidden"
        // or "visible"
        controlsBackArrows: 'faded',

        // Display a presentation progress bar
        progress: true,

        // Display the page number of the current slide
        slideNumber: 'c/t',

        // 'all', 'print', or 'speaker'
        showSlideNumber: 'all',

        // Add the current slide number to the URL hash so that reloading the
        // page/copying the URL will return you to the same slide
        hash: true,

        // Start with 1 for the hash rather than 0
        hashOneBasedIndex: false,

        // Flags if we should monitor the hash and change slides accordingly
        respondToHashChanges: true,

        // Push each slide change to the browser history
        history: true,

        // Enable keyboard shortcuts for navigation
        keyboard: true,

        // Enable the slide overview mode
        overview: true,

        // Disables the default reveal.js slide layout (scaling and centering)
        // so that you can use custom CSS layout
        disableLayout: false,

        // Vertical centering of slides
        center: false,

        // Enables touch navigation on devices with touch input
        touch: true,

        // Loop the presentation
        loop: false,

        // Change the presentation direction to be RTL
        rtl: false,

        // see https://revealjs.com/vertical-slides/#navigation-mode
        navigationMode: 'linear',

        // Randomizes the order of slides each time the presentation loads
        shuffle: false,

        // Turns fragments on and off globally
        fragments: true,

        // Flags whether to include the current fragment in the URL,
        // so that reloading brings you to the same fragment position
        fragmentInURL: false,

        // Flags if the presentation is running in an embedded mode,
        // i.e. contained within a limited portion of the screen
        embedded: false,

        // Flags if we should show a help overlay when the questionmark
        // key is pressed
        help: true,

        // Flags if it should be possible to pause the presentation (blackout)
        pause: true,

        // Flags if speaker notes should be visible to all viewers
        showNotes: false,

        // Global override for autoplaying embedded media (null/true/false)
        autoPlayMedia: null,

        // Global override for preloading lazy-loaded iframes (null/true/false)
        preloadIframes: null,

        // Number of milliseconds between automatically proceeding to the
        // next slide, disabled when set to 0, this value can be overwritten
        // by using a data-autoslide attribute on your slides
        autoSlide: 0,

        // Stop auto-sliding after user input
        autoSlideStoppable: true,

        // Use this method for navigation when auto-sliding
        autoSlideMethod: null,

        // Specify the average time in seconds that you think you will spend
        // presenting each slide. This is used to show a pacing timer in the
        // speaker view
        defaultTiming: null,

        // Enable slide navigation via mouse wheel
        mouseWheel: false,

        // The display mode that will be used to show slides
        display: 'block',

        // Hide cursor if inactive
        hideInactiveCursor: true,

        // Time before the cursor is hidden (in ms)
        hideCursorTime: 5000,

        // Opens links in an iframe preview overlay
        previewLinks: false,

        // Transition style (none/fade/slide/convex/concave/zoom)
        transition: 'none',

        // Transition speed (default/fast/slow)
        transitionSpeed: 'default',

        // Transition style for full page slide backgrounds
        // (none/fade/slide/convex/concave/zoom)
        backgroundTransition: 'none',

        // Number of slides away from the current that are visible
        viewDistance: 3,

        // Number of slides away from the current that are visible on mobile
        // devices. It is advisable to set this to a lower number than
        // viewDistance in order to save resources.
        mobileViewDistance: 2,

        // The "normal" size of the presentation, aspect ratio will be preserved
        // when the presentation is scaled to fit different resolutions. Can be
        // specified using percentage units.
        width: 1050,

        height: 700,

        // Factor of the display size that should remain empty around the content
        margin: 0.1,

        math: {
          mathjax: 'https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.0/MathJax.js',
          config: 'TeX-AMS_HTML-full',
          tex2jax: {
            inlineMath: [['\\(','\\)']],
            displayMath: [['\\[','\\]']],
            balanceBraces: true,
            processEscapes: false,
            processRefs: true,
            processEnvironments: true,
            preview: 'TeX',
            skipTags: ['script','noscript','style','textarea','pre','code'],
            ignoreClass: 'tex2jax_ignore',
            processClass: 'tex2jax_process'
          },
        },

        // reveal.js plugins
        plugins: [QuartoLineHighlight, PdfExport, RevealMenu, RevealChalkboard, QuartoSupport,

          RevealMath,
          RevealNotes,
          RevealSearch,
          RevealZoom
        ]
      });
    </script>
    <script id="quarto-html-after-body" type="application/javascript">
    window.document.addEventListener("DOMContentLoaded", function (event) {
      const toggleBodyColorMode = (bsSheetEl) => {
        const mode = bsSheetEl.getAttribute("data-mode");
        const bodyEl = window.document.querySelector("body");
        if (mode === "dark") {
          bodyEl.classList.add("quarto-dark");
          bodyEl.classList.remove("quarto-light");
        } else {
          bodyEl.classList.add("quarto-light");
          bodyEl.classList.remove("quarto-dark");
        }
      }
      const toggleBodyColorPrimary = () => {
        const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
        if (bsSheetEl) {
          toggleBodyColorMode(bsSheetEl);
        }
      }
      toggleBodyColorPrimary();  
      const tabsets =  window.document.querySelectorAll(".panel-tabset-tabby")
      tabsets.forEach(function(tabset) {
        const tabby = new Tabby('#' + tabset.id);
      });
      const isCodeAnnotation = (el) => {
        for (const clz of el.classList) {
          if (clz.startsWith('code-annotation-')) {                     
            return true;
          }
        }
        return false;
      }
      const clipboard = new window.ClipboardJS('.code-copy-button', {
        text: function(trigger) {
          const codeEl = trigger.previousElementSibling.cloneNode(true);
          for (const childEl of codeEl.children) {
            if (isCodeAnnotation(childEl)) {
              childEl.remove();
            }
          }
          return codeEl.innerText;
        }
      });
      clipboard.on('success', function(e) {
        // button target
        const button = e.trigger;
        // don't keep focus
        button.blur();
        // flash "checked"
        button.classList.add('code-copy-button-checked');
        var currentTitle = button.getAttribute("title");
        button.setAttribute("title", "Copied!");
        let tooltip;
        if (window.bootstrap) {
          button.setAttribute("data-bs-toggle", "tooltip");
          button.setAttribute("data-bs-placement", "left");
          button.setAttribute("data-bs-title", "Copied!");
          tooltip = new bootstrap.Tooltip(button, 
            { trigger: "manual", 
              customClass: "code-copy-button-tooltip",
              offset: [0, -8]});
          tooltip.show();    
        }
        setTimeout(function() {
          if (tooltip) {
            tooltip.hide();
            button.removeAttribute("data-bs-title");
            button.removeAttribute("data-bs-toggle");
            button.removeAttribute("data-bs-placement");
          }
          button.setAttribute("title", currentTitle);
          button.classList.remove('code-copy-button-checked');
        }, 1000);
        // clear code selection
        e.clearSelection();
      });
      function tippyHover(el, contentFn) {
        const config = {
          allowHTML: true,
          content: contentFn,
          maxWidth: 500,
          delay: 100,
          arrow: false,
          appendTo: function(el) {
              return el.closest('section.slide') || el.parentElement;
          },
          interactive: true,
          interactiveBorder: 10,
          theme: 'light-border',
          placement: 'bottom-start'
        };
          config['offset'] = [0,0];
          config['maxWidth'] = 700;
        window.tippy(el, config); 
      }
      const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
      for (var i=0; i<noterefs.length; i++) {
        const ref = noterefs[i];
        tippyHover(ref, function() {
          // use id or data attribute instead here
          let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
          try { href = new URL(href).hash; } catch {}
          const id = href.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          return note.innerHTML;
        });
      }
      const findCites = (el) => {
        const parentEl = el.parentElement;
        if (parentEl) {
          const cites = parentEl.dataset.cites;
          if (cites) {
            return {
              el,
              cites: cites.split(' ')
            };
          } else {
            return findCites(el.parentElement)
          }
        } else {
          return undefined;
        }
      };
      var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
      for (var i=0; i<bibliorefs.length; i++) {
        const ref = bibliorefs[i];
        const citeInfo = findCites(ref);
        if (citeInfo) {
          tippyHover(citeInfo.el, function() {
            var popup = window.document.createElement('div');
            citeInfo.cites.forEach(function(cite) {
              var citeDiv = window.document.createElement('div');
              citeDiv.classList.add('hanging-indent');
              citeDiv.classList.add('csl-entry');
              var biblioDiv = window.document.getElementById('ref-' + cite);
              if (biblioDiv) {
                citeDiv.innerHTML = biblioDiv.innerHTML;
              }
              popup.appendChild(citeDiv);
            });
            return popup.innerHTML;
          });
        }
      }
    });
    </script>
    

</body></html>